<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting Started with ScIsoX • ScIsoX</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Getting Started with ScIsoX">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-#1a5490" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ScIsoX</a>

    <small class="nav-text text-success me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/vignette-01-getting-started.html">Getting Started</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-02-data-import-qc.html">Data Import and QC</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-03-complexity-metrics.html">Complexity Metrics</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-04-visualisation.html">Visualisation Gallery</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-05-coexpression.html">Co-expression Analysis</a></li>
  </ul>
</li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-news" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">News</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-news">
<li><h6 class="dropdown-header" data-toc-skip>Releases</h6></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/v1.1.1">Version 1.1.1</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/V1.1.0">Version 1.1.0</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../news/index.html">Changelog</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ThaddeusWu/ScIsoX/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Getting Started with ScIsoX</h1>
                        <h4 data-toc-skip class="author">Siyuan Wu &amp;
Ulf Schmitz</h4>
            
            <h4 data-toc-skip class="date">2025-08-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ThaddeusWu/ScIsoX/blob/HEAD/vignettes/vignette-01-getting-started.Rmd" class="external-link"><code>vignettes/vignette-01-getting-started.Rmd</code></a></small>
      <div class="d-none name"><code>vignette-01-getting-started.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>ScIsoX is a comprehensive R package for analysing transcriptomic
complexity at single-cell resolution using isoform-level expression
data. This vignette will guide you through the basic workflow of ScIsoX,
from data import to initial complexity analysis.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install from GitHub</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ThaddeusWu/ScIsoX"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ThaddeusWu.github.io/ScIsoX/">ScIsoX</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ThaddeusWu.github.io/ScIsoX/">ScIsoX</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="understanding-the-data-structure">Understanding the Data Structure<a class="anchor" aria-label="anchor" href="#understanding-the-data-structure"></a>
</h2>
<p>ScIsoX works with four main data inputs:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Gene counts matrix</strong>: Gene-level expression (genes ×
cells)</li>
<li>
<strong>Transcript counts matrix</strong>: Isoform-level expression
(transcripts × cells)</li>
<li>
<strong>Transcript information</strong>: Mapping between transcripts
and genes</li>
<li>
<strong>Cell type annotation</strong>: Cell type information</li>
</ol>
<div class="section level3">
<h3 id="loading-example-data">Loading Example Data<a class="anchor" aria-label="anchor" href="#loading-example-data"></a>
</h3>
<p>ScIsoX comes with example data from blood cells:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_counts_blood</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">transcript_counts_blood</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">transcript_info</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">sample2stage</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Examine data dimensions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Gene counts:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">gene_counts_blood</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Gene counts: 55487 205</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Transcript counts:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">transcript_counts_blood</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Transcript counts: 142238 205</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Transcript info:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">transcript_info</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Transcript info: 278305 6</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Cell info:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sample2stage</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Cell info: 203 2</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data-format-requirements">Data Format Requirements<a class="anchor" aria-label="anchor" href="#data-format-requirements"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Gene counts format</span></span>
<span><span class="va">gene_counts_blood</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt;                    AEC-1-AEC-1-1_FKDL202561884-1a-1</span></span>
<span><span class="co">#&gt; ENSMUSG00000000001                                2</span></span>
<span><span class="co">#&gt; ENSMUSG00000000003                                0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000028                                0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000031                             6242</span></span>
<span><span class="co">#&gt; ENSMUSG00000000037                                0</span></span>
<span><span class="co">#&gt;                    AEC-1-AEC-1-2_FKDL202561884-1a-2</span></span>
<span><span class="co">#&gt; ENSMUSG00000000001                                0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000003                                0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000028                               13</span></span>
<span><span class="co">#&gt; ENSMUSG00000000031                            13584</span></span>
<span><span class="co">#&gt; ENSMUSG00000000037                                0</span></span>
<span><span class="co">#&gt;                    AEC-1-AEC-1-3_FKDL202561884-1a-3</span></span>
<span><span class="co">#&gt; ENSMUSG00000000001                               10</span></span>
<span><span class="co">#&gt; ENSMUSG00000000003                                0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000028                                0</span></span>
<span><span class="co">#&gt; ENSMUSG00000000031                             3790</span></span>
<span><span class="co">#&gt; ENSMUSG00000000037                                0</span></span>
<span></span>
<span><span class="co"># Transcript counts format</span></span>
<span><span class="va">transcript_counts_blood</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt;                    AEC-1-AEC-1-1_FKDL202561884-1a-1</span></span>
<span><span class="co">#&gt; ENSMUST00000193812                                0</span></span>
<span><span class="co">#&gt; ENSMUST00000082908                                0</span></span>
<span><span class="co">#&gt; ENSMUST00000162897                                0</span></span>
<span><span class="co">#&gt; ENSMUST00000159265                                0</span></span>
<span><span class="co">#&gt; ENSMUST00000070533                                0</span></span>
<span><span class="co">#&gt;                    AEC-1-AEC-1-2_FKDL202561884-1a-2</span></span>
<span><span class="co">#&gt; ENSMUST00000193812                                0</span></span>
<span><span class="co">#&gt; ENSMUST00000082908                                0</span></span>
<span><span class="co">#&gt; ENSMUST00000162897                                0</span></span>
<span><span class="co">#&gt; ENSMUST00000159265                                0</span></span>
<span><span class="co">#&gt; ENSMUST00000070533                                0</span></span>
<span><span class="co">#&gt;                    AEC-1-AEC-1-3_FKDL202561884-1a-3</span></span>
<span><span class="co">#&gt; ENSMUST00000193812                                0</span></span>
<span><span class="co">#&gt; ENSMUST00000082908                                0</span></span>
<span><span class="co">#&gt; ENSMUST00000162897                                0</span></span>
<span><span class="co">#&gt; ENSMUST00000159265                                0</span></span>
<span><span class="co">#&gt; ENSMUST00000070533                                0</span></span>
<span></span>
<span><span class="co"># Transcript info format</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">transcript_info</span>, <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">#&gt;                         transcript_id   transcript_name            gene_id</span></span>
<span><span class="co">#&gt; ENSMUST00000193812 ENSMUST00000193812 4933401J01Rik-201 ENSMUSG00000102693</span></span>
<span><span class="co">#&gt; ENSMUST00000082908 ENSMUST00000082908       Gm26206-201 ENSMUSG00000064842</span></span>
<span><span class="co">#&gt; ENSMUST00000162897 ENSMUST00000162897          Xkr4-203 ENSMUSG00000051951</span></span>
<span><span class="co">#&gt;                        gene_name                transcript_type      gene_type</span></span>
<span><span class="co">#&gt; ENSMUST00000193812 4933401J01Rik                            TEC            TEC</span></span>
<span><span class="co">#&gt; ENSMUST00000082908       Gm26206                          snRNA          snRNA</span></span>
<span><span class="co">#&gt; ENSMUST00000162897          Xkr4 protein_coding_CDS_not_defined protein_coding</span></span>
<span></span>
<span><span class="co"># Cell info format (optional but recommended)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sample2stage</span>, <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">#&gt;                             sample cell_type</span></span>
<span><span class="co">#&gt; 1 AEC-1-AEC-1-1_FKDL202561884-1a-1       AEC</span></span>
<span><span class="co">#&gt; 2 AEC-1-AEC-1-2_FKDL202561884-1a-2       AEC</span></span>
<span><span class="co">#&gt; 3 AEC-1-AEC-1-3_FKDL202561884-1a-3       AEC</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="quality-control">Quality Control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h2>
<p>Before creating the SCHT structure, it’s important to perform quality
control:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualise genes per cell distribution</span></span>
<span><span class="va">qc_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_genes_per_cell_distribution.html">plot_genes_per_cell_distribution</a></span><span class="op">(</span></span>
<span>  <span class="va">gene_counts_blood</span>,</span>
<span>  plot_type <span class="op">=</span> <span class="st">"hist"</span>,</span>
<span>  return_suggestions <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-01-getting-started_files/figure-html/qc-analysis-1.png" alt="Histogram showing distribution of genes per cell with quality control thresholds" width="672"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Get automated QC recommendations</span></span>
<span><span class="va">qc_recommendations</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/recommend_qc_parameters.html">recommend_qc_parameters</a></span><span class="op">(</span><span class="va">gene_counts_blood</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View different QC strategies</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="st">"Conservative (MAD) strategy:"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Conservative (MAD) strategy:"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">qc_recommendations</span><span class="op">$</span><span class="va">MAD_strategy</span><span class="op">)</span></span>
<span><span class="co">#&gt; $min_genes_per_cell</span></span>
<span><span class="co">#&gt; [1] 980</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $max_genes_per_cell</span></span>
<span><span class="co">#&gt; [1] 13756</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="st">"\nModerate (90% interval) strategy:"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "\nModerate (90% interval) strategy:"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">qc_recommendations</span><span class="op">$</span><span class="va">Interval_90</span><span class="op">)</span></span>
<span><span class="co">#&gt; $min_genes_per_cell</span></span>
<span><span class="co">#&gt; [1] 4106</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $max_genes_per_cell</span></span>
<span><span class="co">#&gt; [1] 10648</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="creating-the-scht-object">Creating the SCHT Object<a class="anchor" aria-label="anchor" href="#creating-the-scht-object"></a>
</h2>
<p>The Single-Cell Hierarchical Tensor (SCHT) is the core data structure
in ScIsoX:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create SCHT with recommended parameters</span></span>
<span><span class="va">scht_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_scht.html">create_scht</a></span><span class="op">(</span></span>
<span>  gene_counts <span class="op">=</span> <span class="va">gene_counts_blood</span>,</span>
<span>  transcript_counts <span class="op">=</span> <span class="va">transcript_counts_blood</span>,</span>
<span>  transcript_info <span class="op">=</span> <span class="va">transcript_info</span>,</span>
<span>  cell_info <span class="op">=</span> <span class="va">sample2stage</span>,</span>
<span>  qc_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    min_genes_per_cell <span class="op">=</span> <span class="fl">4000</span>,</span>
<span>    max_genes_per_cell <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>    min_cells_expressing <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>    min_expr <span class="op">=</span> <span class="fl">1e-6</span></span>
<span>  <span class="op">)</span>,</span>
<span>  n_hvg <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-01-getting-started_files/figure-html/create-scht-1.png" width="672"></p>
<pre><code><span><span class="co">#&gt; Matrix format after log transform:</span></span>
<span><span class="co">#&gt;    class(gene_mat) = dgCMatrix </span></span>
<span><span class="co">#&gt;    dim(gene_mat)   = 17881, 181 </span></span>
<span><span class="co">#&gt;    class(transcript_mat) = dgCMatrix </span></span>
<span><span class="co">#&gt;    dim(transcript_mat)   = 142238, 181</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; === Sparsity Analysis for Current Dataset ===</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 1. Original Transcript Matrix:</span></span>
<span><span class="co">#&gt;   Number of genes: 55,487</span></span>
<span><span class="co">#&gt;   Number of isoforms: 142,238</span></span>
<span><span class="co">#&gt;   Number of cells: 205</span></span>
<span><span class="co">#&gt;   Non-zero elements: 2,457,789</span></span>
<span><span class="co">#&gt;   Zero elements: 26,701,001</span></span>
<span><span class="co">#&gt;   Total elements: 29,158,790</span></span>
<span><span class="co">#&gt;   Sparsity: 91.57%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 2. Filtered Transcript Matrix (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Number of genes: 1,936</span></span>
<span><span class="co">#&gt;   Number of isoforms: 7,327</span></span>
<span><span class="co">#&gt;   Number of cells: 181</span></span>
<span><span class="co">#&gt;   Non-zero elements: 73,057</span></span>
<span><span class="co">#&gt;   Zero elements: 1,253,130</span></span>
<span><span class="co">#&gt;   Total elements: 1,326,187</span></span>
<span><span class="co">#&gt;   Sparsity: 94.49%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 3. SCHT Structure (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Non-zero elements: 73,057</span></span>
<span><span class="co">#&gt;   Zero elements: 154,780</span></span>
<span><span class="co">#&gt;   Total elements: 227,837</span></span>
<span><span class="co">#&gt;   Sparsity: 67.93%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 4. Naive 3D Tensor (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Required dimensions: 1,936 x 21 x 181</span></span>
<span><span class="co">#&gt;   Non-zero elements: 73,057</span></span>
<span><span class="co">#&gt;   Zero elements: 7,285,679</span></span>
<span><span class="co">#&gt;   Total elements: 7,358,736</span></span>
<span><span class="co">#&gt;   Sparsity: 99.01%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 5. Zero Elements Avoided by SCHT:</span></span>
<span><span class="co">#&gt;   vs Original Matrix: 26,546,221</span></span>
<span><span class="co">#&gt;   vs Filtered Matrix: 1,098,350</span></span>
<span><span class="co">#&gt;   vs Naive 3D Tensor: 7,130,899</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 6. Memory Efficiency Summary:</span></span>
<span><span class="co">#&gt;   SCHT uses only 0.78% of original matrix size</span></span>
<span><span class="co">#&gt;   SCHT uses only 17.18% of filtered matrix size</span></span>
<span><span class="co">#&gt;   SCHT uses only 3.10% of naive tensor size</span></span>
<span></span>
<span><span class="co"># View SCHT structure</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">scht_obj</span><span class="op">)</span></span>
<span><span class="co">#&gt; Integrated SCHT Object</span></span>
<span><span class="co">#&gt; =====================</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Original SCHT:</span></span>
<span><span class="co">#&gt; Single-Cell Hierarchical Tensor (SCHT) object</span></span>
<span><span class="co">#&gt; Number of genes: 1936</span></span>
<span><span class="co">#&gt; Number of cells: 181</span></span>
<span><span class="co">#&gt; Total number of transcripts: 7327</span></span>
<span><span class="co">#&gt; Mean isoforms per gene: 3.78</span></span>
<span><span class="co">#&gt; Overall sparsity: 67.9 %</span></span>
<span><span class="co">#&gt; Matrix storage type: matrix</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Cell type-specific analysis:</span></span>
<span><span class="co">#&gt; Number of cell types: 7</span></span>
<span><span class="co">#&gt; Cell Types: AEC, HEC, T1_pre_HSC, T2_pre_HSC, E12, E14, Adult_HSC </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Created: 2025-08-01 03:12:35</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">scht_obj</span><span class="op">)</span></span>
<span><span class="co">#&gt; Integrated SCHT Summary</span></span>
<span><span class="co">#&gt; ======================</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Original SCHT Summary:</span></span>
<span><span class="co">#&gt; ----------------------</span></span>
<span><span class="co">#&gt; SCHT Object Summary:</span></span>
<span><span class="co">#&gt; --------------------</span></span>
<span><span class="co">#&gt;   Cells: 181</span></span>
<span><span class="co">#&gt;   Genes: 1936</span></span>
<span><span class="co">#&gt;   Total transcripts: 7327</span></span>
<span><span class="co">#&gt;   Mean isoforms: 3.78</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Preprocessing Info:</span></span>
<span><span class="co">#&gt;   HVGs selected: 3000</span></span>
<span><span class="co">#&gt;   QC-filtered genes: 37606</span></span>
<span><span class="co">#&gt;   QC-filtered transcripts: 0</span></span>
<span><span class="co">#&gt;   QC-filtered cells: 24</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Data characteristics:</span></span>
<span><span class="co">#&gt;   Sparsity: 67.9%</span></span>
<span><span class="co">#&gt;   Created: 2025-08-01 03:12:31</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Cell type-specific Summary:</span></span>
<span><span class="co">#&gt; ----------------------</span></span>
<span><span class="co">#&gt;    cell_type n_genes n_cells</span></span>
<span><span class="co">#&gt; 1        AEC    1385      18</span></span>
<span><span class="co">#&gt; 2        HEC    1383      24</span></span>
<span><span class="co">#&gt; 3 T1_pre_HSC    1638      26</span></span>
<span><span class="co">#&gt; 4 T2_pre_HSC    1531      38</span></span>
<span><span class="co">#&gt; 5        E12    1361      21</span></span>
<span><span class="co">#&gt; 6        E14    1262      32</span></span>
<span><span class="co">#&gt; 7  Adult_HSC    1346      44</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Performance metrics:</span></span>
<span><span class="co">#&gt;   Total processing time: 31.91 seconds (0.53 minutes)</span></span>
<span><span class="co">#&gt;   Memory utilised: 183.40 MB</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Note: The actual number of cells for each gene may vary from the shown n_cells,</span></span>
<span><span class="co">#&gt; as cells with no expression for specific genes are removed from their respective matrices.</span></span>
<span><span class="co">#&gt; This cell-wise filtering is performed independently for each gene to maintain data quality</span></span>
<span><span class="co">#&gt; and avoid spurious zero expressions in the cell type-specific analyses.</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="calculating-complexity-metrics">Calculating Complexity Metrics<a class="anchor" aria-label="anchor" href="#calculating-complexity-metrics"></a>
</h2>
<p>ScIsoX calculates seven core metrics to quantify transcriptomic
complexity:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate all complexity metrics</span></span>
<span><span class="va">tc_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_isoform_complexity_metrics.html">calculate_isoform_complexity_metrics</a></span><span class="op">(</span></span>
<span>  <span class="va">scht_obj</span>, </span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View results structure</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">)</span></span>
<span><span class="co">#&gt; Isoform Complexity Analysis Result</span></span>
<span><span class="co">#&gt; =================================</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Total genes analysed: 1936</span></span>
<span><span class="co">#&gt; Classification thresholds determined by: data-driven modelling</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Top complexity classes:</span></span>
<span><span class="co">#&gt;   1. Low Diversity + Low Specificity: 1024 genes (52.9%)</span></span>
<span><span class="co">#&gt;   2. High Diversity + High Specificity: 319 genes (16.5%)</span></span>
<span><span class="co">#&gt;   3. Low Diversity + High Specificity: 317 genes (16.4%)</span></span>
<span><span class="co">#&gt;   4. High Diversity + Low Specificity: 276 genes (14.3%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Call summary() for detailed statistics</span></span>
<span></span>
<span><span class="co"># Summary statistics</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">)</span></span>
<span><span class="co">#&gt; Isoform Complexity Analysis Summary:</span></span>
<span><span class="co">#&gt; Total genes analysed: 1936</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; NA Value Statistics (representing biologically meaningful cases):</span></span>
<span><span class="co">#&gt;   - Intra-cellular Isoform Diversity   :    0 genes (0.0%) - Occurs when genes have only one expressed isoform, indicating lack of alternative splicing</span></span>
<span><span class="co">#&gt;   - Inter-cellular Isoform Diversity   :    0 genes (0.0%) - Occurs when only one isoform is expressed across the entire cell population</span></span>
<span><span class="co">#&gt;   - Intra-cell-type Heterogeneity      :  163 genes (8.4%) - Occurs when a cell type has insufficient cells expressing the gene (&lt; 3 cells)</span></span>
<span><span class="co">#&gt;   - Inter-cell-type Specificity        :    0 genes (0.0%) - Occurs for genes expressed in only a single cell type (high cell type-specificity)</span></span>
<span><span class="co">#&gt;   - Intra-cell-type Heterogeneity Variability:  492 genes (25.4%) - Occurs when a gene is expressed in fewer than 2 cell types with sufficient data</span></span>
<span><span class="co">#&gt;   - Inter-cell-type Difference Variability:  137 genes (7.1%) - Occurs when there are insufficient pairwise differences between cell types to calculate variability</span></span>
<span><span class="co">#&gt;   - Cell-type-specific Co-expression Variability:  184 genes (9.5%) - Occurs when a gene lacks sufficient cell type-specific expression data to compare co-expression patterns</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Classification Distribution Across Complexity Dimensions:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Intra-cellular Isoform Diversity:</span></span>
<span><span class="co">#&gt;   - Strong Isoform Co-expression            :  704 genes (36.4%)</span></span>
<span><span class="co">#&gt;   - Weak Isoform Co-expression              : 1232 genes (63.6%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Inter-cellular Isoform Diversity:</span></span>
<span><span class="co">#&gt;   - High Isoform Diversity                  :  595 genes (30.7%)</span></span>
<span><span class="co">#&gt;   - Low Isoform Diversity                   : 1341 genes (69.3%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Intra-cell-type Heterogeneity:</span></span>
<span><span class="co">#&gt;   - High Cellular Heterogeneity             :  819 genes (42.3%)</span></span>
<span><span class="co">#&gt;   - Low Cellular Heterogeneity              :  954 genes (49.3%)</span></span>
<span><span class="co">#&gt;   - Unclassified                            :  163 genes (8.4%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Inter-cell-type Specificity:</span></span>
<span><span class="co">#&gt;   - Cell-Type-Independent Isoform Expression: 1300 genes (67.1%)</span></span>
<span><span class="co">#&gt;   - Cell-Type-Specific Isoform Expression   :  636 genes (32.9%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Intra-cell-type Heterogeneity Variability:</span></span>
<span><span class="co">#&gt;   - Consistent Heterogeneity Across Cell Types:  892 genes (46.1%)</span></span>
<span><span class="co">#&gt;   - Insufficient Cell Type Data             :  492 genes (25.4%)</span></span>
<span><span class="co">#&gt;   - Variable Heterogeneity Across Cell Types:  552 genes (28.5%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Inter-cell-type Difference Variability:</span></span>
<span><span class="co">#&gt;   - High Cell-Type Distinctions             :  943 genes (48.7%)</span></span>
<span><span class="co">#&gt;   - Insufficient Difference Data            :  137 genes (7.1%)</span></span>
<span><span class="co">#&gt;   - Low Cell-Type Distinctions              :  856 genes (44.2%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Cell-type-specific Co-expression Variability:</span></span>
<span><span class="co">#&gt;   - Cell-Type-Adaptive Co-expression        :  596 genes (30.8%)</span></span>
<span><span class="co">#&gt;   - Cell-Type-Consistent Co-expression      : 1156 genes (59.7%)</span></span>
<span><span class="co">#&gt;   - Insufficient Data                       :  184 genes (9.5%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Traditional Complexity Categories:</span></span>
<span><span class="co">#&gt;   - High Diversity + High Specificity       :  319 genes (16.5%)</span></span>
<span><span class="co">#&gt;   - High Diversity + Low Specificity        :  276 genes (14.3%)</span></span>
<span><span class="co">#&gt;   - Low Diversity + High Specificity        :  317 genes (16.4%)</span></span>
<span><span class="co">#&gt;   - Low Diversity + Low Specificity         : 1024 genes (52.9%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Cell-Type-Specific Genes:</span></span>
<span><span class="co">#&gt;   - Single cell-type genes                  :   23 genes (1.2%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Core Metrics Statistics:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; |Metric                                       |  Mean| Median|    SD|  Min|   Max| Threshold| NA_Percent|</span></span>
<span><span class="co">#&gt; |:--------------------------------------------|-----:|------:|-----:|----:|-----:|---------:|----------:|</span></span>
<span><span class="co">#&gt; |Intra-cellular Isoform Diversity             | 0.351|  0.321| 0.270| 0.00| 0.991|     0.456|      0.000|</span></span>
<span><span class="co">#&gt; |Inter-cellular Isoform Diversity             | 0.591|  0.630| 0.257| 0.01| 1.000|     0.758|      0.000|</span></span>
<span><span class="co">#&gt; |Intra-cell-type Heterogeneity                | 0.363|  0.367| 0.230| 0.00| 1.000|     0.389|      8.419|</span></span>
<span><span class="co">#&gt; |Inter-cell-type Specificity                  | 0.401|  0.382| 0.230| 0.00| 1.000|     0.497|      0.000|</span></span>
<span><span class="co">#&gt; |Intra-cell-type Heterogeneity Variability    | 0.770|  0.620| 0.625| 0.00| 2.646|     0.832|     25.413|</span></span>
<span><span class="co">#&gt; |Inter-cell-type Difference Variability       | 0.646|  0.567| 0.342| 0.00| 1.620|     0.552|      7.076|</span></span>
<span><span class="co">#&gt; |Cell-type-specific Co-expression Variability | 1.032|  0.972| 0.702| 0.00| 2.646|     1.389|      9.504|</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Performance metrics:</span></span>
<span><span class="co">#&gt;   Processing time: 83.82 seconds (1.40 minutes)</span></span>
<span><span class="co">#&gt;   Memory utilised: 975.63 MB</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="basic-visualisation">Basic Visualisation<a class="anchor" aria-label="anchor" href="#basic-visualisation"></a>
</h2>
<p>Visualise the complexity landscape:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create complexity landscape plot</span></span>
<span><span class="fu"><a href="../reference/plot_tc_landscape.html">plot_tc_landscape</a></span><span class="op">(</span></span>
<span>  <span class="va">tc_results</span>,</span>
<span>  n_label <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-01-getting-started_files/figure-html/basic-vis-1.png" alt="Complexity landscape plot showing relationship between intra-cellular and inter-cellular isoform diversity" width="1440"></p>
</div>
<div class="section level2">
<h2 id="finding-genes-of-interest">Finding Genes of Interest<a class="anchor" aria-label="anchor" href="#finding-genes-of-interest"></a>
</h2>
<p>Identify genes with specific complexity patterns:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find highly complex genes</span></span>
<span><span class="va">complex_genes</span> <span class="op">&lt;-</span> <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">[</span></span>
<span>  <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">intra_cellular_isoform_diversity_class</span> <span class="op">==</span> <span class="st">"Strong Isoform Co-expression"</span> <span class="op">&amp;</span></span>
<span>  <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">inter_cellular_isoform_diversity_class</span> <span class="op">==</span> <span class="st">"High Isoform Diversity"</span>,</span>
<span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Found"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">complex_genes</span><span class="op">)</span>, <span class="st">"highly complex genes"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Found 384 highly complex genes"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">complex_genes</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"n_isoforms"</span>, </span>
<span>                       <span class="st">"intra_cellular_isoform_diversity"</span>,</span>
<span>                       <span class="st">"inter_cellular_isoform_diversity"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;            gene n_isoforms intra_cellular_isoform_diversity</span></span>
<span><span class="co">#&gt; Cd177     Cd177          2                        0.8339576</span></span>
<span><span class="co">#&gt; S100a8   S100a8          2                        0.7979806</span></span>
<span><span class="co">#&gt; S100a9   S100a9          2                        0.7408728</span></span>
<span><span class="co">#&gt; Il7         Il7          4                        0.7749878</span></span>
<span><span class="co">#&gt; Nup62cl Nup62cl          2                        0.8197578</span></span>
<span><span class="co">#&gt; Crb3       Crb3          4                        0.6215834</span></span>
<span><span class="co">#&gt;         inter_cellular_isoform_diversity</span></span>
<span><span class="co">#&gt; Cd177                          0.9015444</span></span>
<span><span class="co">#&gt; S100a8                         0.9042768</span></span>
<span><span class="co">#&gt; S100a9                         0.9078688</span></span>
<span><span class="co">#&gt; Il7                            0.8055326</span></span>
<span><span class="co">#&gt; Nup62cl                        0.9041507</span></span>
<span><span class="co">#&gt; Crb3                           0.9335333</span></span>
<span></span>
<span><span class="co"># Find genes with cell type-specific patterns</span></span>
<span><span class="va">ct_specific</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/select_genes_of_interest.html">select_genes_of_interest</a></span><span class="op">(</span></span>
<span>    <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span>,</span>
<span>    category <span class="op">=</span> <span class="st">"Cell-Type-Specific Isoform Expression"</span>,</span>
<span>    top_n <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ct_specific</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "Cd177"    "S100a8"   "S100a9"   "Shtn1"    "Ceacam10" "Oas3"    </span></span>
<span><span class="co">#&gt;  [7] "Il7"      "Nup62cl"  "Cideb"    "Ctss"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="next-steps">Next Steps<a class="anchor" aria-label="anchor" href="#next-steps"></a>
</h2>
<p>This vignette covered the basic workflow of ScIsoX. For more advanced
analyses, see:</p>
<ul>
<li>
<strong>Data Import and QC</strong>: Detailed quality control
strategies</li>
<li>
<strong>Complexity Metrics</strong>: In-depth explanation of all
seven metrics</li>
<li>
<strong>Visualisation Gallery</strong>: All 13+ visualisation
functions</li>
<li>
<strong>Co-expression Analysis</strong>: Isoform relationships and
switching</li>
</ul>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.3 (2025-02-28)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Sequoia 15.5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Australia/Brisbane</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ScIsoX_1.1.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] bitops_1.0-9                gridExtra_2.3              </span></span>
<span><span class="co">#&gt;   [3] rlang_1.1.6                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [5] clue_0.3-66                 GetoptLong_1.0.5           </span></span>
<span><span class="co">#&gt;   [7] ggridges_0.5.6              matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt;   [9] compiler_4.4.3              png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [11] systemfonts_1.2.1           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;  [13] pkgconfig_2.0.3             shape_1.4.6.1              </span></span>
<span><span class="co">#&gt;  [15] crayon_1.5.3                fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [17] XVector_0.46.0              labeling_0.4.3             </span></span>
<span><span class="co">#&gt;  [19] promises_1.3.3              Rsamtools_2.22.0           </span></span>
<span><span class="co">#&gt;  [21] rmarkdown_2.29              UCSC.utils_1.2.0           </span></span>
<span><span class="co">#&gt;  [23] ragg_1.4.0                  purrr_1.1.0                </span></span>
<span><span class="co">#&gt;  [25] xfun_0.52                   zlibbioc_1.52.0            </span></span>
<span><span class="co">#&gt;  [27] cachem_1.1.0                GenomeInfoDb_1.42.3        </span></span>
<span><span class="co">#&gt;  [29] jsonlite_2.0.0              progress_1.2.3             </span></span>
<span><span class="co">#&gt;  [31] later_1.4.2                 DelayedArray_0.32.0        </span></span>
<span><span class="co">#&gt;  [33] BiocParallel_1.40.0         parallel_4.4.3             </span></span>
<span><span class="co">#&gt;  [35] prettyunits_1.2.0           cluster_2.1.8.1            </span></span>
<span><span class="co">#&gt;  [37] R6_2.6.1                    bslib_0.9.0                </span></span>
<span><span class="co">#&gt;  [39] RColorBrewer_1.1-3          rtracklayer_1.66.0         </span></span>
<span><span class="co">#&gt;  [41] car_3.1-3                   GenomicRanges_1.58.0       </span></span>
<span><span class="co">#&gt;  [43] jquerylib_0.1.4             diptest_0.77-1             </span></span>
<span><span class="co">#&gt;  [45] Rcpp_1.1.0                  SummarizedExperiment_1.36.0</span></span>
<span><span class="co">#&gt;  [47] iterators_1.0.14            knitr_1.50                 </span></span>
<span><span class="co">#&gt;  [49] IRanges_2.40.1              httpuv_1.6.16              </span></span>
<span><span class="co">#&gt;  [51] Matrix_1.7-3                tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt;  [53] viridis_0.6.5               rstudioapi_0.17.1          </span></span>
<span><span class="co">#&gt;  [55] abind_1.4-8                 yaml_2.3.10                </span></span>
<span><span class="co">#&gt;  [57] miniUI_0.1.1.1              doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [59] codetools_0.2-20            curl_6.4.0                 </span></span>
<span><span class="co">#&gt;  [61] lattice_0.22-6              tibble_3.3.0               </span></span>
<span><span class="co">#&gt;  [63] plyr_1.8.9                  withr_3.0.2                </span></span>
<span><span class="co">#&gt;  [65] shiny_1.11.1                Biobase_2.66.0             </span></span>
<span><span class="co">#&gt;  [67] evaluate_1.0.4              moments_0.14.1             </span></span>
<span><span class="co">#&gt;  [69] desc_1.4.3                  circlize_0.4.16            </span></span>
<span><span class="co">#&gt;  [71] mclust_6.1.1                Biostrings_2.74.1          </span></span>
<span><span class="co">#&gt;  [73] pillar_1.11.0               MatrixGenerics_1.18.1      </span></span>
<span><span class="co">#&gt;  [75] carData_3.0-5               DT_0.33                    </span></span>
<span><span class="co">#&gt;  [77] foreach_1.5.2               stats4_4.4.3               </span></span>
<span><span class="co">#&gt;  [79] plotly_4.11.0               generics_0.1.4             </span></span>
<span><span class="co">#&gt;  [81] RCurl_1.98-1.16             S4Vectors_0.44.0           </span></span>
<span><span class="co">#&gt;  [83] hms_1.1.3                   ggplot2_3.5.2              </span></span>
<span><span class="co">#&gt;  [85] scales_1.4.0                xtable_1.8-4               </span></span>
<span><span class="co">#&gt;  [87] glue_1.8.0                  lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;  [89] tools_4.4.3                 BiocIO_1.16.0              </span></span>
<span><span class="co">#&gt;  [91] data.table_1.17.8           ggradar_0.2                </span></span>
<span><span class="co">#&gt;  [93] GenomicAlignments_1.42.0    fs_1.6.6                   </span></span>
<span><span class="co">#&gt;  [95] XML_3.99-0.18               cowplot_1.2.0              </span></span>
<span><span class="co">#&gt;  [97] grid_4.4.3                  tidyr_1.3.1                </span></span>
<span><span class="co">#&gt;  [99] colorspace_2.1-1            patchwork_1.3.0            </span></span>
<span><span class="co">#&gt; [101] GenomeInfoDbData_1.2.13     restfulr_0.0.15            </span></span>
<span><span class="co">#&gt; [103] Formula_1.2-5               cli_3.6.5                  </span></span>
<span><span class="co">#&gt; [105] textshaping_1.0.0           S4Arrays_1.6.0             </span></span>
<span><span class="co">#&gt; [107] viridisLite_0.4.2           ComplexHeatmap_2.22.0      </span></span>
<span><span class="co">#&gt; [109] dplyr_1.1.4                 gtable_0.3.6               </span></span>
<span><span class="co">#&gt; [111] sass_0.4.10                 digest_0.6.37              </span></span>
<span><span class="co">#&gt; [113] BiocGenerics_0.52.0         ggrepel_0.9.6              </span></span>
<span><span class="co">#&gt; [115] SparseArray_1.6.2           rjson_0.2.23               </span></span>
<span><span class="co">#&gt; [117] htmlwidgets_1.6.4           farver_2.1.2               </span></span>
<span><span class="co">#&gt; [119] htmltools_0.5.8.1           pkgdown_2.1.3              </span></span>
<span><span class="co">#&gt; [121] lifecycle_1.0.4             httr_1.4.7                 </span></span>
<span><span class="co">#&gt; [123] mime_0.13                   GlobalOptions_0.1.2        </span></span>
<span><span class="co">#&gt; [125] ggExtra_0.10.1              MASS_7.3-65</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Siyuan Wu, Ulf Schmitz<br> Site built with <a href="https://pkgdown.r-lib.org" class="external-link">pkgdown</a> 2.1.1</p>
</div>

<div class="pkgdown-footer-right">
  <p><strong>Package License</strong>: <a href="https://opensource.org/licenses/MIT" class="external-link">MIT</a><br><br><strong>Example Data Attribution</strong>:<br> The example datasets included in this package are derived from:<br> Wang et al. (2022). Single-cell multiomics defines tolerogenic extrathymic Aire-expressing populations. <em>Science Advances</em> 8(1):eabg5369<br> Licensed under <a href="https://creativecommons.org/licenses/by/4.0/" class="external-link">CC BY 4.0</a><br> We gratefully acknowledge the authors for making their data publicly available.</p>
</div>

    </footer>
</div>





  </body>
</html>
