<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Data Import and Quality Control • ScIsoX</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Data Import and Quality Control">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-#1a5490" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ScIsoX</a>

    <small class="nav-text text-success me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/vignette-01-getting-started.html">Getting Started</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-02-data-import-qc.html">Data Import and QC</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-03-complexity-metrics.html">Complexity Metrics</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-04-visualisation.html">Visualisation Gallery</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-05-coexpression.html">Co-expression Analysis</a></li>
  </ul>
</li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-news" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">News</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-news">
<li><h6 class="dropdown-header" data-toc-skip>Releases</h6></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/v1.1.1">Version 1.1.1</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/V1.1.0">Version 1.1.0</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../news/index.html">Changelog</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ThaddeusWu/ScIsoX/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Data Import and Quality Control</h1>
                        <h4 data-toc-skip class="author">Siyuan Wu &amp;
Ulf Schmitz</h4>
            
            <h4 data-toc-skip class="date">2025-08-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ThaddeusWu/ScIsoX/blob/HEAD/vignettes/vignette-02-data-import-qc.Rmd" class="external-link"><code>vignettes/vignette-02-data-import-qc.Rmd</code></a></small>
      <div class="d-none name"><code>vignette-02-data-import-qc.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette covers the essential steps for importing data and
performing quality control (QC) in ScIsoX. Proper QC is crucial for
obtaining reliable complexity metrics.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ThaddeusWu.github.io/ScIsoX/">ScIsoX</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-requirements">Data Requirements<a class="anchor" aria-label="anchor" href="#data-requirements"></a>
</h2>
<p>ScIsoX requires three main inputs:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Gene counts matrix</strong>: Gene-level expression (genes ×
cells)</li>
<li>
<strong>Transcript counts matrix</strong>: Isoform-level expression
(transcripts × cells)<br>
</li>
<li>
<strong>Transcript information</strong>: Mapping between transcripts
and genes</li>
</ol>
<div class="section level3">
<h3 id="loading-example-data">Loading Example Data<a class="anchor" aria-label="anchor" href="#loading-example-data"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_counts_blood</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">transcript_counts_blood</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">transcript_info</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">sample2stage</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check dimensions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Gene counts:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">gene_counts_blood</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Gene counts: 55487 205</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Transcript counts:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">transcript_counts_blood</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Transcript counts: 142238 205</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Number of cells with metadata:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sample2stage</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Number of cells with metadata: 203</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="quality-control-visualisation">Quality Control Visualisation<a class="anchor" aria-label="anchor" href="#quality-control-visualisation"></a>
</h2>
<p>Before creating the SCHT structure, examine your data quality:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualise genes per cell distribution</span></span>
<span><span class="va">qc_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_genes_per_cell_distribution.html">plot_genes_per_cell_distribution</a></span><span class="op">(</span></span>
<span>  <span class="va">gene_counts_blood</span>,</span>
<span>  plot_type <span class="op">=</span> <span class="st">"hist"</span>,</span>
<span>  return_suggestions <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-02-data-import-qc_files/figure-html/qc-viz-1.png" alt="Histogram of genes per cell distribution with suggested quality control thresholds" width="768"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># The plot shows distribution and suggests thresholds</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">qc_plot</span><span class="op">$</span><span class="va">suggestions</span><span class="op">)</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="automated-qc-recommendations">Automated QC Recommendations<a class="anchor" aria-label="anchor" href="#automated-qc-recommendations"></a>
</h2>
<p>ScIsoX can automatically recommend QC parameters based on your
data:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get automated recommendations</span></span>
<span><span class="va">qc_rec</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/recommend_qc_parameters.html">recommend_qc_parameters</a></span><span class="op">(</span><span class="va">gene_counts_blood</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-02-data-import-qc_files/figure-html/auto-qc-1.png" width="768"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># View different strategies</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"=== Conservative Strategy (MAD-based) ===\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; === Conservative Strategy (MAD-based) ===</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">qc_rec</span><span class="op">$</span><span class="va">MAD_strategy</span><span class="op">)</span></span>
<span><span class="co">#&gt; $min_genes_per_cell</span></span>
<span><span class="co">#&gt; [1] 980</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $max_genes_per_cell</span></span>
<span><span class="co">#&gt; [1] 13756</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\n=== Moderate Strategy (90% interval) ===\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; === Moderate Strategy (90% interval) ===</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">qc_rec</span><span class="op">$</span><span class="va">Interval_90</span><span class="op">)</span></span>
<span><span class="co">#&gt; $min_genes_per_cell</span></span>
<span><span class="co">#&gt; [1] 4106</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $max_genes_per_cell</span></span>
<span><span class="co">#&gt; [1] 10648</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\n=== Lenient Strategy (80% interval) ===\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; === Lenient Strategy (80% interval) ===</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">qc_rec</span><span class="op">$</span><span class="va">Interval_80</span><span class="op">)</span></span>
<span><span class="co">#&gt; $min_genes_per_cell</span></span>
<span><span class="co">#&gt; [1] 4474</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $max_genes_per_cell</span></span>
<span><span class="co">#&gt; [1] 9663</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="creating-transcript-information">Creating Transcript Information<a class="anchor" aria-label="anchor" href="#creating-transcript-information"></a>
</h2>
<p>If you need to generate transcript information from a GTF file:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create transcript info from GTF</span></span>
<span><span class="va">transcript_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_transcript_info.html">create_transcript_info</a></span><span class="op">(</span></span>
<span>  gtf_file <span class="op">=</span> <span class="st">"path/to/annotation.gtf"</span>,</span>
<span>  remove_version <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View the structure</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">transcript_info</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="generating-gene-counts">Generating Gene Counts<a class="anchor" aria-label="anchor" href="#generating-gene-counts"></a>
</h2>
<p>If you only have transcript counts, you can generate gene counts:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Generate gene counts from transcript counts</span></span>
<span><span class="co"># Note: Row names must be in GeneName-TranscriptID format</span></span>
<span><span class="va">gene_counts_generated</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_gene_counts.html">generate_gene_counts</a></span><span class="op">(</span><span class="va">transcript_counts_blood</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare with original</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Generated genes:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">gene_counts_generated</span><span class="op">$</span><span class="va">gene_counts</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Generated genes: 142238</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Original genes:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">gene_counts_blood</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Original genes: 55487</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="creating-the-scht-object">Creating the SCHT Object<a class="anchor" aria-label="anchor" href="#creating-the-scht-object"></a>
</h2>
<p>With QC parameters selected, create the SCHT structure:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Use moderate QC strategy</span></span>
<span><span class="va">qc_params</span> <span class="op">&lt;-</span> <span class="va">qc_rec</span><span class="op">$</span><span class="va">Interval_90</span></span>
<span><span class="va">qc_params</span><span class="op">$</span><span class="va">min_cells_expressing</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">qc_params</span><span class="op">$</span><span class="va">min_expr</span> <span class="op">&lt;-</span> <span class="fl">1e-6</span></span>
<span></span>
<span><span class="co"># Create SCHT</span></span>
<span><span class="va">scht_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_scht.html">create_scht</a></span><span class="op">(</span></span>
<span>  gene_counts <span class="op">=</span> <span class="va">gene_counts_blood</span>,</span>
<span>  transcript_counts <span class="op">=</span> <span class="va">transcript_counts_blood</span>,</span>
<span>  transcript_info <span class="op">=</span> <span class="va">transcript_info</span>,</span>
<span>  cell_info <span class="op">=</span> <span class="va">sample2stage</span>,</span>
<span>  qc_params <span class="op">=</span> <span class="va">qc_params</span>,</span>
<span>  n_hvg <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-02-data-import-qc_files/figure-html/create-scht-1.png" width="768"></p>
<pre><code><span><span class="co">#&gt; Matrix format after log transform:</span></span>
<span><span class="co">#&gt;    class(gene_mat) = dgCMatrix </span></span>
<span><span class="co">#&gt;    dim(gene_mat)   = 17881, 183 </span></span>
<span><span class="co">#&gt;    class(transcript_mat) = dgCMatrix </span></span>
<span><span class="co">#&gt;    dim(transcript_mat)   = 142238, 183</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; === Sparsity Analysis for Current Dataset ===</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 1. Original Transcript Matrix:</span></span>
<span><span class="co">#&gt;   Number of genes: 55,487</span></span>
<span><span class="co">#&gt;   Number of isoforms: 142,238</span></span>
<span><span class="co">#&gt;   Number of cells: 205</span></span>
<span><span class="co">#&gt;   Non-zero elements: 2,457,789</span></span>
<span><span class="co">#&gt;   Zero elements: 26,701,001</span></span>
<span><span class="co">#&gt;   Total elements: 29,158,790</span></span>
<span><span class="co">#&gt;   Sparsity: 91.57%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 2. Filtered Transcript Matrix (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Number of genes: 1,938</span></span>
<span><span class="co">#&gt;   Number of isoforms: 7,380</span></span>
<span><span class="co">#&gt;   Number of cells: 183</span></span>
<span><span class="co">#&gt;   Non-zero elements: 75,064</span></span>
<span><span class="co">#&gt;   Zero elements: 1,275,476</span></span>
<span><span class="co">#&gt;   Total elements: 1,350,540</span></span>
<span><span class="co">#&gt;   Sparsity: 94.44%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 3. SCHT Structure (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Non-zero elements: 75,064</span></span>
<span><span class="co">#&gt;   Zero elements: 162,725</span></span>
<span><span class="co">#&gt;   Total elements: 237,789</span></span>
<span><span class="co">#&gt;   Sparsity: 68.43%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 4. Naive 3D Tensor (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Required dimensions: 1,938 x 21 x 183</span></span>
<span><span class="co">#&gt;   Non-zero elements: 75,064</span></span>
<span><span class="co">#&gt;   Zero elements: 7,372,670</span></span>
<span><span class="co">#&gt;   Total elements: 7,447,734</span></span>
<span><span class="co">#&gt;   Sparsity: 98.99%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 5. Zero Elements Avoided by SCHT:</span></span>
<span><span class="co">#&gt;   vs Original Matrix: 26,538,276</span></span>
<span><span class="co">#&gt;   vs Filtered Matrix: 1,112,751</span></span>
<span><span class="co">#&gt;   vs Naive 3D Tensor: 7,209,945</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 6. Memory Efficiency Summary:</span></span>
<span><span class="co">#&gt;   SCHT uses only 0.82% of original matrix size</span></span>
<span><span class="co">#&gt;   SCHT uses only 17.61% of filtered matrix size</span></span>
<span><span class="co">#&gt;   SCHT uses only 3.19% of naive tensor size</span></span>
<span></span>
<span><span class="co"># View the result</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">scht_obj</span><span class="op">)</span></span>
<span><span class="co">#&gt; Integrated SCHT Summary</span></span>
<span><span class="co">#&gt; ======================</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Original SCHT Summary:</span></span>
<span><span class="co">#&gt; ----------------------</span></span>
<span><span class="co">#&gt; SCHT Object Summary:</span></span>
<span><span class="co">#&gt; --------------------</span></span>
<span><span class="co">#&gt;   Cells: 183</span></span>
<span><span class="co">#&gt;   Genes: 1938</span></span>
<span><span class="co">#&gt;   Total transcripts: 7380</span></span>
<span><span class="co">#&gt;   Mean isoforms: 3.81</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Preprocessing Info:</span></span>
<span><span class="co">#&gt;   HVGs selected: 3000</span></span>
<span><span class="co">#&gt;   QC-filtered genes: 37606</span></span>
<span><span class="co">#&gt;   QC-filtered transcripts: 0</span></span>
<span><span class="co">#&gt;   QC-filtered cells: 22</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Data characteristics:</span></span>
<span><span class="co">#&gt;   Sparsity: 68.4%</span></span>
<span><span class="co">#&gt;   Created: 2025-08-01 03:14:43</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Cell type-specific Summary:</span></span>
<span><span class="co">#&gt; ----------------------</span></span>
<span><span class="co">#&gt;    cell_type n_genes n_cells</span></span>
<span><span class="co">#&gt; 1        AEC    1388      18</span></span>
<span><span class="co">#&gt; 2        HEC    1548      24</span></span>
<span><span class="co">#&gt; 3 T1_pre_HSC    1676      26</span></span>
<span><span class="co">#&gt; 4 T2_pre_HSC    1568      38</span></span>
<span><span class="co">#&gt; 5        E12    1359      21</span></span>
<span><span class="co">#&gt; 6        E14    1263      32</span></span>
<span><span class="co">#&gt; 7  Adult_HSC    1267      44</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Performance metrics:</span></span>
<span><span class="co">#&gt;   Total processing time: 35.41 seconds (0.59 minutes)</span></span>
<span><span class="co">#&gt;   Memory utilised: 237.00 MB</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Note: The actual number of cells for each gene may vary from the shown n_cells,</span></span>
<span><span class="co">#&gt; as cells with no expression for specific genes are removed from their respective matrices.</span></span>
<span><span class="co">#&gt; This cell-wise filtering is performed independently for each gene to maintain data quality</span></span>
<span><span class="co">#&gt; and avoid spurious zero expressions in the cell type-specific analyses.</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="generating-qc-report">Generating QC Report<a class="anchor" aria-label="anchor" href="#generating-qc-report"></a>
</h2>
<p>For a comprehensive QC analysis, generate an HTML report:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Generate detailed QC report</span></span>
<span><span class="fu"><a href="../reference/generate_qc_report.html">generate_qc_report</a></span><span class="op">(</span></span>
<span>  gene_counts <span class="op">=</span> <span class="va">gene_counts_blood</span>,</span>
<span>  output_file <span class="op">=</span> <span class="st">"qc_report.html"</span>,</span>
<span>  dataset_name <span class="op">=</span> <span class="st">"Blood Cells Analysis"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="best-practices">Best Practices<a class="anchor" aria-label="anchor" href="#best-practices"></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<strong>Always visualise your data</strong> before setting QC
parameters</li>
<li>
<strong>Start with automated recommendations</strong> then adjust
based on your experimental design</li>
<li>
<strong>Consider your sequencing depth</strong> when setting
thresholds:
<ul>
<li>Deep sequencing: Higher gene count thresholds</li>
<li>Shallow sequencing: Lower thresholds</li>
</ul>
</li>
<li>
<strong>Document your QC choices</strong> for reproducibility</li>
</ol>
</div>
<div class="section level2">
<h2 id="next-steps">Next Steps<a class="anchor" aria-label="anchor" href="#next-steps"></a>
</h2>
<p>After QC and SCHT creation, you’re ready to: - Calculate complexity
metrics (see “Complexity Metrics” vignette) - Explore visualisations
(see “Visualisation Gallery” vignette) - Perform co-expression analysis
(see “Co-expression Analysis” vignette)</p>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.3 (2025-02-28)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Sequoia 15.5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Australia/Brisbane</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ScIsoX_1.1.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] bitops_1.0-9                gridExtra_2.3              </span></span>
<span><span class="co">#&gt;   [3] rlang_1.1.6                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [5] clue_0.3-66                 GetoptLong_1.0.5           </span></span>
<span><span class="co">#&gt;   [7] ggridges_0.5.6              matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt;   [9] compiler_4.4.3              png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [11] systemfonts_1.2.1           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;  [13] pkgconfig_2.0.3             shape_1.4.6.1              </span></span>
<span><span class="co">#&gt;  [15] crayon_1.5.3                fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [17] XVector_0.46.0              promises_1.3.3             </span></span>
<span><span class="co">#&gt;  [19] Rsamtools_2.22.0            rmarkdown_2.29             </span></span>
<span><span class="co">#&gt;  [21] UCSC.utils_1.2.0            ragg_1.4.0                 </span></span>
<span><span class="co">#&gt;  [23] purrr_1.1.0                 xfun_0.52                  </span></span>
<span><span class="co">#&gt;  [25] zlibbioc_1.52.0             cachem_1.1.0               </span></span>
<span><span class="co">#&gt;  [27] GenomeInfoDb_1.42.3         jsonlite_2.0.0             </span></span>
<span><span class="co">#&gt;  [29] progress_1.2.3              later_1.4.2                </span></span>
<span><span class="co">#&gt;  [31] DelayedArray_0.32.0         BiocParallel_1.40.0        </span></span>
<span><span class="co">#&gt;  [33] parallel_4.4.3              prettyunits_1.2.0          </span></span>
<span><span class="co">#&gt;  [35] cluster_2.1.8.1             R6_2.6.1                   </span></span>
<span><span class="co">#&gt;  [37] bslib_0.9.0                 RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt;  [39] rtracklayer_1.66.0          car_3.1-3                  </span></span>
<span><span class="co">#&gt;  [41] GenomicRanges_1.58.0        jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [43] diptest_0.77-1              Rcpp_1.1.0                 </span></span>
<span><span class="co">#&gt;  [45] SummarizedExperiment_1.36.0 iterators_1.0.14           </span></span>
<span><span class="co">#&gt;  [47] knitr_1.50                  IRanges_2.40.1             </span></span>
<span><span class="co">#&gt;  [49] httpuv_1.6.16               Matrix_1.7-3               </span></span>
<span><span class="co">#&gt;  [51] tidyselect_1.2.1            viridis_0.6.5              </span></span>
<span><span class="co">#&gt;  [53] rstudioapi_0.17.1           abind_1.4-8                </span></span>
<span><span class="co">#&gt;  [55] yaml_2.3.10                 miniUI_0.1.1.1             </span></span>
<span><span class="co">#&gt;  [57] doParallel_1.0.17           codetools_0.2-20           </span></span>
<span><span class="co">#&gt;  [59] curl_6.4.0                  lattice_0.22-6             </span></span>
<span><span class="co">#&gt;  [61] tibble_3.3.0                plyr_1.8.9                 </span></span>
<span><span class="co">#&gt;  [63] shiny_1.11.1                Biobase_2.66.0             </span></span>
<span><span class="co">#&gt;  [65] evaluate_1.0.4              moments_0.14.1             </span></span>
<span><span class="co">#&gt;  [67] desc_1.4.3                  circlize_0.4.16            </span></span>
<span><span class="co">#&gt;  [69] mclust_6.1.1                Biostrings_2.74.1          </span></span>
<span><span class="co">#&gt;  [71] pillar_1.11.0               MatrixGenerics_1.18.1      </span></span>
<span><span class="co">#&gt;  [73] carData_3.0-5               DT_0.33                    </span></span>
<span><span class="co">#&gt;  [75] foreach_1.5.2               stats4_4.4.3               </span></span>
<span><span class="co">#&gt;  [77] plotly_4.11.0               generics_0.1.4             </span></span>
<span><span class="co">#&gt;  [79] RCurl_1.98-1.16             S4Vectors_0.44.0           </span></span>
<span><span class="co">#&gt;  [81] hms_1.1.3                   ggplot2_3.5.2              </span></span>
<span><span class="co">#&gt;  [83] scales_1.4.0                xtable_1.8-4               </span></span>
<span><span class="co">#&gt;  [85] glue_1.8.0                  lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;  [87] tools_4.4.3                 BiocIO_1.16.0              </span></span>
<span><span class="co">#&gt;  [89] data.table_1.17.8           ggradar_0.2                </span></span>
<span><span class="co">#&gt;  [91] GenomicAlignments_1.42.0    fs_1.6.6                   </span></span>
<span><span class="co">#&gt;  [93] XML_3.99-0.18               cowplot_1.2.0              </span></span>
<span><span class="co">#&gt;  [95] grid_4.4.3                  tidyr_1.3.1                </span></span>
<span><span class="co">#&gt;  [97] colorspace_2.1-1            patchwork_1.3.0            </span></span>
<span><span class="co">#&gt;  [99] GenomeInfoDbData_1.2.13     restfulr_0.0.15            </span></span>
<span><span class="co">#&gt; [101] Formula_1.2-5               cli_3.6.5                  </span></span>
<span><span class="co">#&gt; [103] textshaping_1.0.0           S4Arrays_1.6.0             </span></span>
<span><span class="co">#&gt; [105] viridisLite_0.4.2           ComplexHeatmap_2.22.0      </span></span>
<span><span class="co">#&gt; [107] dplyr_1.1.4                 gtable_0.3.6               </span></span>
<span><span class="co">#&gt; [109] sass_0.4.10                 digest_0.6.37              </span></span>
<span><span class="co">#&gt; [111] BiocGenerics_0.52.0         ggrepel_0.9.6              </span></span>
<span><span class="co">#&gt; [113] SparseArray_1.6.2           rjson_0.2.23               </span></span>
<span><span class="co">#&gt; [115] htmlwidgets_1.6.4           farver_2.1.2               </span></span>
<span><span class="co">#&gt; [117] htmltools_0.5.8.1           pkgdown_2.1.3              </span></span>
<span><span class="co">#&gt; [119] lifecycle_1.0.4             httr_1.4.7                 </span></span>
<span><span class="co">#&gt; [121] mime_0.13                   GlobalOptions_0.1.2        </span></span>
<span><span class="co">#&gt; [123] ggExtra_0.10.1              MASS_7.3-65</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Siyuan Wu, Ulf Schmitz<br> Site built with <a href="https://pkgdown.r-lib.org" class="external-link">pkgdown</a> 2.1.1</p>
</div>

<div class="pkgdown-footer-right">
  <p><strong>Package License</strong>: <a href="https://opensource.org/licenses/MIT" class="external-link">MIT</a><br><br><strong>Example Data Attribution</strong>:<br> The example datasets included in this package are derived from:<br> Wang et al. (2022). Single-cell multiomics defines tolerogenic extrathymic Aire-expressing populations. <em>Science Advances</em> 8(1):eabg5369<br> Licensed under <a href="https://creativecommons.org/licenses/by/4.0/" class="external-link">CC BY 4.0</a><br> We gratefully acknowledge the authors for making their data publicly available.</p>
</div>

    </footer>
</div>





  </body>
</html>
