<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Visualisation Gallery and Interpretation • ScIsoX</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Visualisation Gallery and Interpretation">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-#1a5490" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ScIsoX</a>

    <small class="nav-text text-success me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/vignette-01-getting-started.html">Getting Started</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-02-data-import-qc.html">Data Import and QC</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-03-complexity-metrics.html">Complexity Metrics</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-04-visualisation.html">Visualisation Gallery</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-05-coexpression.html">Co-expression Analysis</a></li>
  </ul>
</li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-news" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">News</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-news">
<li><h6 class="dropdown-header" data-toc-skip>Releases</h6></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/v1.1.1">Version 1.1.1</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/V1.1.0">Version 1.1.0</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../news/index.html">Changelog</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ThaddeusWu/ScIsoX/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Visualisation Gallery and Interpretation</h1>
                        <h4 data-toc-skip class="author">Siyuan Wu &amp;
Ulf Schmitz</h4>
            
            <h4 data-toc-skip class="date">2025-08-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ThaddeusWu/ScIsoX/blob/HEAD/vignettes/vignette-04-visualisation.Rmd" class="external-link"><code>vignettes/vignette-04-visualisation.Rmd</code></a></small>
      <div class="d-none name"><code>vignette-04-visualisation.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>ScIsoX provides 13+ specialised visualisation functions to explore
transcriptomic complexity from different perspectives. This vignette
demonstrates each visualisation type with interpretation guidelines.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ThaddeusWu.github.io/ScIsoX/">ScIsoX</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load example data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_counts_blood</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">transcript_counts_blood</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">transcript_info</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">sample2stage</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create SCHT object</span></span>
<span><span class="va">scht_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_scht.html">create_scht</a></span><span class="op">(</span></span>
<span>  gene_counts <span class="op">=</span> <span class="va">gene_counts_blood</span>,</span>
<span>  transcript_counts <span class="op">=</span> <span class="va">transcript_counts_blood</span>,</span>
<span>  transcript_info <span class="op">=</span> <span class="va">transcript_info</span>,</span>
<span>  cell_info <span class="op">=</span> <span class="va">sample2stage</span>,</span>
<span>  qc_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    min_genes_per_cell <span class="op">=</span> <span class="fl">4000</span>,</span>
<span>    max_genes_per_cell <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>    min_cells_expressing <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>    min_expr <span class="op">=</span> <span class="fl">1e-6</span></span>
<span>  <span class="op">)</span>,</span>
<span>  n_hvg <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-04-visualisation_files/figure-html/load-package-1.png" width="960"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; === Sparsity Analysis for Current Dataset ===</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 1. Original Transcript Matrix:</span></span>
<span><span class="co">#&gt;   Number of genes: 55,487</span></span>
<span><span class="co">#&gt;   Number of isoforms: 142,238</span></span>
<span><span class="co">#&gt;   Number of cells: 205</span></span>
<span><span class="co">#&gt;   Non-zero elements: 2,457,789</span></span>
<span><span class="co">#&gt;   Zero elements: 26,701,001</span></span>
<span><span class="co">#&gt;   Total elements: 29,158,790</span></span>
<span><span class="co">#&gt;   Sparsity: 91.57%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 2. Filtered Transcript Matrix (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Number of genes: 1,936</span></span>
<span><span class="co">#&gt;   Number of isoforms: 7,327</span></span>
<span><span class="co">#&gt;   Number of cells: 181</span></span>
<span><span class="co">#&gt;   Non-zero elements: 73,057</span></span>
<span><span class="co">#&gt;   Zero elements: 1,253,130</span></span>
<span><span class="co">#&gt;   Total elements: 1,326,187</span></span>
<span><span class="co">#&gt;   Sparsity: 94.49%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 3. SCHT Structure (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Non-zero elements: 73,057</span></span>
<span><span class="co">#&gt;   Zero elements: 154,780</span></span>
<span><span class="co">#&gt;   Total elements: 227,837</span></span>
<span><span class="co">#&gt;   Sparsity: 67.93%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 4. Naive 3D Tensor (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Required dimensions: 1,936 x 21 x 181</span></span>
<span><span class="co">#&gt;   Non-zero elements: 73,057</span></span>
<span><span class="co">#&gt;   Zero elements: 7,285,679</span></span>
<span><span class="co">#&gt;   Total elements: 7,358,736</span></span>
<span><span class="co">#&gt;   Sparsity: 99.01%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 5. Zero Elements Avoided by SCHT:</span></span>
<span><span class="co">#&gt;   vs Original Matrix: 26,546,221</span></span>
<span><span class="co">#&gt;   vs Filtered Matrix: 1,098,350</span></span>
<span><span class="co">#&gt;   vs Naive 3D Tensor: 7,130,899</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 6. Memory Efficiency Summary:</span></span>
<span><span class="co">#&gt;   SCHT uses only 0.78% of original matrix size</span></span>
<span><span class="co">#&gt;   SCHT uses only 17.18% of filtered matrix size</span></span>
<span><span class="co">#&gt;   SCHT uses only 3.10% of naive tensor size</span></span>
<span></span>
<span><span class="co"># Calculate complexity metrics</span></span>
<span><span class="va">tc_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_isoform_complexity_metrics.html">calculate_isoform_complexity_metrics</a></span><span class="op">(</span><span class="va">scht_obj</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="complexity-landscape-visualisations">Complexity Landscape Visualisations<a class="anchor" aria-label="anchor" href="#complexity-landscape-visualisations"></a>
</h2>
<div class="section level3">
<h3 id="complexity-landscape">1. Complexity Landscape<a class="anchor" aria-label="anchor" href="#complexity-landscape"></a>
</h3>
<p><strong>Purpose</strong>: Visualise the relationship between
complexity metrics in a 2D scatter plot.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tc_landscape.html">plot_tc_landscape</a></span><span class="op">(</span></span>
<span>  <span class="va">tc_results</span>,</span>
<span>  n_label <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  label_direction <span class="op">=</span> <span class="st">"top"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-04-visualisation_files/figure-html/landscape-1.png" alt="Complexity landscape scatter plot showing genes in 2D complexity space" width="1440"></p>
<p><strong>Interpretation</strong>: - <strong>Top-right</strong>: Genes
with high complexity in both dimensions - <strong>Bottom-left</strong>:
Simple genes with limited isoform diversity - <strong>Top-left</strong>:
Genes with cell-specific isoform expression -
<strong>Bottom-right</strong>: Genes with constitutive multi-isoform
expression</p>
</div>
<div class="section level3">
<h3 id="density-contour-plot">2. Density Contour Plot<a class="anchor" aria-label="anchor" href="#density-contour-plot"></a>
</h3>
<p><strong>Purpose</strong>: Show the density distribution of genes in
complexity space.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tc_density.html">plot_tc_density</a></span><span class="op">(</span></span>
<span>  <span class="va">tc_results</span>,</span>
<span>  use_thresholds <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-04-visualisation_files/figure-html/density-1.png" alt="Density contour plot showing gene distribution in complexity space" width="960"></p>
<p><strong>Interpretation</strong>: - Contour lines indicate gene
density - Multiple peaks suggest distinct gene populations - Useful for
identifying common complexity patterns</p>
</div>
<div class="section level3">
<h3 id="diversity-comparison">3. Diversity Comparison<a class="anchor" aria-label="anchor" href="#diversity-comparison"></a>
</h3>
<p><strong>Purpose</strong>: Compare the distributions of different
diversity metrics.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_diversity_comparison.html">plot_diversity_comparison</a></span><span class="op">(</span></span>
<span>  <span class="va">tc_results</span>,</span>
<span>  label_top <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-04-visualisation_files/figure-html/diversity-comp-1.png" alt="Comparison plot of different diversity metrics" width="960"></p>
</div>
<div class="section level3">
<h3 id="complexity-ridge-plots">4. Complexity Ridge Plots<a class="anchor" aria-label="anchor" href="#complexity-ridge-plots"></a>
</h3>
<p><strong>Purpose</strong>: Show distributions of complexity metrics
across categories.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># For global view</span></span>
<span><span class="fu"><a href="../reference/plot_complexity_ridges.html">plot_complexity_ridges</a></span><span class="op">(</span></span>
<span>  <span class="va">tc_results</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"global"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-04-visualisation_files/figure-html/ridges-1.png" alt="Ridge plots showing complexity metric distributions" width="960"></p>
</div>
<div class="section level3">
<h3 id="threshold-visualisations">5. Threshold Visualisations<a class="anchor" aria-label="anchor" href="#threshold-visualisations"></a>
</h3>
<p><strong>Purpose</strong>: Understand how metrics are classified into
low/medium/high categories.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Note: This function expects threshold plots object, not tc_results</span></span>
<span><span class="co"># Skip this example as it requires different input</span></span>
<span><span class="fu"><a href="../reference/plot_threshold_visualisations.html">plot_threshold_visualisations</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">threshold_plots</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-04-visualisation_files/figure-html/threshold-1.png" alt="Threshold visualisation plots showing metric classification" width="1920"></p>
<pre><code><span><span class="co">#&gt; TableGrob (2 x 1) "arrange": 2 grobs</span></span>
<span><span class="co">#&gt;   z     cells    name                grob</span></span>
<span><span class="co">#&gt; 1 1 (2-2,1-1) arrange     gtable[arrange]</span></span>
<span><span class="co">#&gt; 2 2 (1-1,1-1) arrange text[GRID.text.493]</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="isoform-specific-visualisations">Isoform-Specific Visualisations<a class="anchor" aria-label="anchor" href="#isoform-specific-visualisations"></a>
</h2>
<div class="section level3">
<h3 id="isoform-co-expression-heatmap">6. Isoform Co-expression Heatmap<a class="anchor" aria-label="anchor" href="#isoform-co-expression-heatmap"></a>
</h3>
<p><strong>Purpose</strong>: Visualise correlation patterns between
isoforms of a gene.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/plot_isoform_coexpression.html">plot_isoform_coexpression</a></span><span class="op">(</span></span>
<span>  <span class="va">scht_obj</span>,</span>
<span>  gene <span class="op">=</span> <span class="st">"Irf8"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"pearson"</span>,</span>
<span>  display_numbers <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-04-visualisation_files/figure-html/coexp-heatmap-1.png" alt="Heatmap showing correlation patterns between isoforms" width="768"></p>
<p><strong>Interpretation</strong>: - Red: Positive correlation
(co-expressed isoforms) - Blue: Negative correlation (mutually exclusive
isoforms) - Clustering reveals isoform groups</p>
</div>
<div class="section level3">
<h3 id="isoform-usage-profile">7. Isoform Usage Profile<a class="anchor" aria-label="anchor" href="#isoform-usage-profile"></a>
</h3>
<p><strong>Purpose</strong>: Show the relative usage of isoforms across
cell types.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_isoform_profile.html">plot_isoform_profile</a></span><span class="op">(</span></span>
<span>  <span class="va">scht_obj</span>,</span>
<span>  gene <span class="op">=</span> <span class="st">"Irf8"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-04-visualisation_files/figure-html/iso-profile-1.png" alt="Bar plot showing relative isoform usage across cell types" width="960"></p>
</div>
<div class="section level3">
<h3 id="isoform-transitions">8. Isoform Transitions<a class="anchor" aria-label="anchor" href="#isoform-transitions"></a>
</h3>
<p><strong>Purpose</strong>: Visualise how isoform usage changes across
cell types or states.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_isoform_transitions.html">plot_isoform_transitions</a></span><span class="op">(</span></span>
<span>  <span class="va">scht_obj</span>,</span>
<span>  gene <span class="op">=</span> <span class="st">"Atl1"</span>,</span>
<span>  cell_type_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AEC"</span>,<span class="st">"HEC"</span>,<span class="st">"T1_pre_HSC"</span>,<span class="st">"T2_pre_HSC"</span>,<span class="st">"E12"</span>,<span class="st">"E14"</span>,<span class="st">"Adult_HSC"</span> <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-04-visualisation_files/figure-html/iso-transitions-1.png" alt="Line plot showing isoform usage transitions across cell types" width="960"></p>
</div>
</div>
<div class="section level2">
<h2 id="multi-dimensional-comparisons">Multi-dimensional Comparisons<a class="anchor" aria-label="anchor" href="#multi-dimensional-comparisons"></a>
</h2>
<div class="section level3">
<h3 id="complexity-radar-chart">9. Complexity Radar Chart<a class="anchor" aria-label="anchor" href="#complexity-radar-chart"></a>
</h3>
<p><strong>Purpose</strong>: Compare multiple complexity metrics for
selected genes.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Select interesting genes</span></span>
<span><span class="va">complex_genes</span> <span class="op">&lt;-</span> <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">[</span></span>
<span>  <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">cell_type_coexpression_variability_class</span> <span class="op">==</span> <span class="st">"Cell-Type-Consistent Co-expression"</span>,</span>
<span><span class="op">]</span><span class="op">$</span><span class="va">gene</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">complex_genes</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="../reference/plot_complexity_radar.html">plot_complexity_radar</a></span><span class="op">(</span></span>
<span>    <span class="va">tc_results</span>,</span>
<span>    genes <span class="op">=</span> <span class="va">complex_genes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">complex_genes</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="vignette-04-visualisation_files/figure-html/radar-1.png" alt="Radar chart comparing complexity metrics for multiple genes" width="1440"></p>
</div>
<div class="section level3">
<h3 id="single-gene-radar-across-cell-types">10. Single Gene Radar Across Cell Types<a class="anchor" aria-label="anchor" href="#single-gene-radar-across-cell-types"></a>
</h3>
<p><strong>Purpose</strong>: Show how a gene’s complexity varies across
cell types.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/plot_single_gene_radar_cell_type.html">plot_single_gene_radar_cell_type</a></span><span class="op">(</span></span>
<span>  <span class="va">tc_results</span>,</span>
<span>  gene_name <span class="op">=</span> <span class="st">"Irf8"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-04-visualisation_files/figure-html/single-radar-1.png" alt="Radar chart showing single gene complexity across cell types" width="1440"></p>
</div>
<div class="section level3">
<h3 id="multiple-genes-radar-comparison">11. Multiple Genes Radar Comparison<a class="anchor" aria-label="anchor" href="#multiple-genes-radar-comparison"></a>
</h3>
<p><strong>Purpose</strong>: Compare multiple genes across cell types
simultaneously.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_compare_multiple_genes_radar_cell_type.html">plot_compare_multiple_genes_radar_cell_type</a></span><span class="op">(</span></span>
<span>  <span class="va">tc_results</span>,</span>
<span>  gene_names <span class="op">=</span> <span class="va">complex_genes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">complex_genes</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  scale_type <span class="op">=</span> <span class="st">"per_cell_type"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-04-visualisation_files/figure-html/multi-radar-1.png" alt="Multiple radar charts comparing genes across cell types" width="1440"></p>
</div>
</div>
<div class="section level2">
<h2 id="interpretation-guidelines">Interpretation Guidelines<a class="anchor" aria-label="anchor" href="#interpretation-guidelines"></a>
</h2>
<div class="section level3">
<h3 id="for-complexity-landscapes">For Complexity Landscapes:<a class="anchor" aria-label="anchor" href="#for-complexity-landscapes"></a>
</h3>
<ol style="list-style-type: decimal">
<li>
<strong>Gene positioning</strong>: Location indicates relative
complexity</li>
<li>
<strong>Clustering patterns</strong>: Groups of genes with similar
profiles</li>
<li>
<strong>Outliers</strong>: Often biologically interesting genes</li>
<li>
<strong>Threshold lines</strong>: Help identify significantly
complex genes</li>
</ol>
</div>
<div class="section level3">
<h3 id="for-isoform-visualisations">For Isoform Visualisations:<a class="anchor" aria-label="anchor" href="#for-isoform-visualisations"></a>
</h3>
<ol style="list-style-type: decimal">
<li>
<strong>Co-expression patterns</strong>: Indicate functional
relationships</li>
<li>
<strong>Cell type specificity</strong>: Suggests regulatory
mechanisms</li>
<li>
<strong>Transition patterns</strong>: May indicate differentiation
paths</li>
</ol>
</div>
<div class="section level3">
<h3 id="for-multi-dimensional-views">For Multi-dimensional Views:<a class="anchor" aria-label="anchor" href="#for-multi-dimensional-views"></a>
</h3>
<ol style="list-style-type: decimal">
<li>
<strong>Radar charts</strong>: Compare overall complexity
profiles</li>
<li>
<strong>Heatmaps</strong>: Identify patterns across many
genes/conditions</li>
<li>
<strong>Ridge plots</strong>: Understand distribution shapes</li>
</ol>
</div>
</div>
<div class="section level2">
<h2 id="customisation-options">Customisation Options<a class="anchor" aria-label="anchor" href="#customisation-options"></a>
</h2>
<p>Most visualisation functions support customisation:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example: Customised landscape plot</span></span>
<span><span class="fu"><a href="../reference/plot_tc_landscape.html">plot_tc_landscape</a></span><span class="op">(</span></span>
<span>  <span class="va">tc_results</span>,</span>
<span>  x_metric <span class="op">=</span> <span class="st">"intra_cellular_isoform_diversity"</span>,  <span class="co"># Change axes</span></span>
<span>  y_metric <span class="op">=</span> <span class="st">"intra_cell_type_heterogeneity_variability"</span>,</span>
<span>  highlight_genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Gene1"</span>, <span class="st">"Gene2"</span><span class="op">)</span>,          <span class="co"># Highlight specific genes</span></span>
<span>  n_label <span class="op">=</span> <span class="fl">20</span>,                                   <span class="co"># Label more genes</span></span>
<span>  point_transparency <span class="op">=</span> <span class="fl">0.7</span>                        <span class="co"># Adjust transparency</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="saving-plots">Saving Plots<a class="anchor" aria-label="anchor" href="#saving-plots"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># High-resolution plots for publication</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a></span><span class="op">(</span><span class="st">"complexity_analysis.pdf"</span>, width <span class="op">=</span> <span class="fl">12</span>, height <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_tc_landscape.html">plot_tc_landscape</a></span><span class="op">(</span><span class="va">tc_results</span>, n_label <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># PNG for presentations</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span><span class="op">(</span><span class="st">"complexity_landscape.png"</span>, width <span class="op">=</span> <span class="fl">1200</span>, height <span class="op">=</span> <span class="fl">1000</span>, res <span class="op">=</span> <span class="fl">150</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_tc_landscape.html">plot_tc_landscape</a></span><span class="op">(</span><span class="va">tc_results</span>, n_label <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="best-practices">Best Practices<a class="anchor" aria-label="anchor" href="#best-practices"></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<strong>Start with overview plots</strong> (landscape, density) to
understand overall patterns</li>
<li>
<strong>Zoom in on interesting genes</strong> using isoform-specific
plots</li>
<li>
<strong>Use comparative plots</strong> when analysing multiple
conditions</li>
<li>
<strong>Adjust parameters</strong> based on your dataset size and
complexity</li>
<li>
<strong>Consider colour-blind friendly palettes</strong> for
publication figures</li>
</ol>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.3 (2025-02-28)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Sequoia 15.5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Australia/Brisbane</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ggplot2_3.5.2 ScIsoX_1.1.1 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3          rstudioapi_0.17.1          </span></span>
<span><span class="co">#&gt;   [3] jsonlite_2.0.0              shape_1.4.6.1              </span></span>
<span><span class="co">#&gt;   [5] magrittr_2.0.3              farver_2.1.2               </span></span>
<span><span class="co">#&gt;   [7] rmarkdown_2.29              GlobalOptions_0.1.2        </span></span>
<span><span class="co">#&gt;   [9] fs_1.6.6                    BiocIO_1.16.0              </span></span>
<span><span class="co">#&gt;  [11] zlibbioc_1.52.0             ragg_1.4.0                 </span></span>
<span><span class="co">#&gt;  [13] vctrs_0.6.5                 Rsamtools_2.22.0           </span></span>
<span><span class="co">#&gt;  [15] RCurl_1.98-1.16             forcats_1.0.0              </span></span>
<span><span class="co">#&gt;  [17] htmltools_0.5.8.1           S4Arrays_1.6.0             </span></span>
<span><span class="co">#&gt;  [19] progress_1.2.3              curl_6.4.0                 </span></span>
<span><span class="co">#&gt;  [21] SparseArray_1.6.2           Formula_1.2-5              </span></span>
<span><span class="co">#&gt;  [23] sass_0.4.10                 bslib_0.9.0                </span></span>
<span><span class="co">#&gt;  [25] htmlwidgets_1.6.4           desc_1.4.3                 </span></span>
<span><span class="co">#&gt;  [27] plyr_1.8.9                  plotly_4.11.0              </span></span>
<span><span class="co">#&gt;  [29] cachem_1.1.0                GenomicAlignments_1.42.0   </span></span>
<span><span class="co">#&gt;  [31] mime_0.13                   lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt;  [33] iterators_1.0.14            pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [35] Matrix_1.7-3                R6_2.6.1                   </span></span>
<span><span class="co">#&gt;  [37] fastmap_1.2.0               GenomeInfoDbData_1.2.13    </span></span>
<span><span class="co">#&gt;  [39] MatrixGenerics_1.18.1       shiny_1.11.1               </span></span>
<span><span class="co">#&gt;  [41] clue_0.3-66                 digest_0.6.37              </span></span>
<span><span class="co">#&gt;  [43] colorspace_2.1-1            patchwork_1.3.0            </span></span>
<span><span class="co">#&gt;  [45] S4Vectors_0.44.0            textshaping_1.0.0          </span></span>
<span><span class="co">#&gt;  [47] GenomicRanges_1.58.0        labeling_0.4.3             </span></span>
<span><span class="co">#&gt;  [49] httr_1.4.7                  abind_1.4-8                </span></span>
<span><span class="co">#&gt;  [51] mgcv_1.9-1                  compiler_4.4.3             </span></span>
<span><span class="co">#&gt;  [53] withr_3.0.2                 doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [55] BiocParallel_1.40.0         carData_3.0-5              </span></span>
<span><span class="co">#&gt;  [57] viridis_0.6.5               MASS_7.3-65                </span></span>
<span><span class="co">#&gt;  [59] DelayedArray_0.32.0         rjson_0.2.23               </span></span>
<span><span class="co">#&gt;  [61] tools_4.4.3                 ggradar_0.2                </span></span>
<span><span class="co">#&gt;  [63] httpuv_1.6.16               glue_1.8.0                 </span></span>
<span><span class="co">#&gt;  [65] restfulr_0.0.15             nlme_3.1-167               </span></span>
<span><span class="co">#&gt;  [67] promises_1.3.3              grid_4.4.3                 </span></span>
<span><span class="co">#&gt;  [69] cluster_2.1.8.1             generics_0.1.4             </span></span>
<span><span class="co">#&gt;  [71] isoband_0.2.7               gtable_0.3.6               </span></span>
<span><span class="co">#&gt;  [73] tidyr_1.3.1                 data.table_1.17.8          </span></span>
<span><span class="co">#&gt;  [75] hms_1.1.3                   car_3.1-3                  </span></span>
<span><span class="co">#&gt;  [77] XVector_0.46.0              BiocGenerics_0.52.0        </span></span>
<span><span class="co">#&gt;  [79] ggrepel_0.9.6               foreach_1.5.2              </span></span>
<span><span class="co">#&gt;  [81] pillar_1.11.0               ggExtra_0.10.1             </span></span>
<span><span class="co">#&gt;  [83] later_1.4.2                 circlize_0.4.16            </span></span>
<span><span class="co">#&gt;  [85] splines_4.4.3               dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [87] moments_0.14.1              lattice_0.22-6             </span></span>
<span><span class="co">#&gt;  [89] rtracklayer_1.66.0          tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt;  [91] ComplexHeatmap_2.22.0       Biostrings_2.74.1          </span></span>
<span><span class="co">#&gt;  [93] miniUI_0.1.1.1              knitr_1.50                 </span></span>
<span><span class="co">#&gt;  [95] gridExtra_2.3               IRanges_2.40.1             </span></span>
<span><span class="co">#&gt;  [97] SummarizedExperiment_1.36.0 stats4_4.4.3               </span></span>
<span><span class="co">#&gt;  [99] xfun_0.52                   Biobase_2.66.0             </span></span>
<span><span class="co">#&gt; [101] diptest_0.77-1              matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt; [103] DT_0.33                     UCSC.utils_1.2.0           </span></span>
<span><span class="co">#&gt; [105] lazyeval_0.2.2              yaml_2.3.10                </span></span>
<span><span class="co">#&gt; [107] evaluate_1.0.4              codetools_0.2-20           </span></span>
<span><span class="co">#&gt; [109] tibble_3.3.0                cli_3.6.5                  </span></span>
<span><span class="co">#&gt; [111] xtable_1.8-4                systemfonts_1.2.1          </span></span>
<span><span class="co">#&gt; [113] jquerylib_0.1.4             Rcpp_1.1.0                 </span></span>
<span><span class="co">#&gt; [115] GenomeInfoDb_1.42.3         png_0.1-8                  </span></span>
<span><span class="co">#&gt; [117] XML_3.99-0.18               parallel_4.4.3             </span></span>
<span><span class="co">#&gt; [119] pkgdown_2.1.3               prettyunits_1.2.0          </span></span>
<span><span class="co">#&gt; [121] mclust_6.1.1                bitops_1.0-9               </span></span>
<span><span class="co">#&gt; [123] viridisLite_0.4.2           scales_1.4.0               </span></span>
<span><span class="co">#&gt; [125] ggridges_0.5.6              purrr_1.1.0                </span></span>
<span><span class="co">#&gt; [127] crayon_1.5.3                GetoptLong_1.0.5           </span></span>
<span><span class="co">#&gt; [129] rlang_1.1.6                 cowplot_1.2.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Siyuan Wu, Ulf Schmitz<br> Site built with <a href="https://pkgdown.r-lib.org" class="external-link">pkgdown</a> 2.1.1</p>
</div>

<div class="pkgdown-footer-right">
  <p><strong>Package License</strong>: <a href="https://opensource.org/licenses/MIT" class="external-link">MIT</a><br><br><strong>Example Data Attribution</strong>:<br> The example datasets included in this package are derived from:<br> Wang et al. (2022). Single-cell multiomics defines tolerogenic extrathymic Aire-expressing populations. <em>Science Advances</em> 8(1):eabg5369<br> Licensed under <a href="https://creativecommons.org/licenses/by/4.0/" class="external-link">CC BY 4.0</a><br> We gratefully acknowledge the authors for making their data publicly available.</p>
</div>

    </footer>
</div>





  </body>
</html>
