<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Understanding Transcriptomic Complexity Metrics • ScIsoX</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Understanding Transcriptomic Complexity Metrics">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-#1a5490" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ScIsoX</a>

    <small class="nav-text text-success me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/vignette-01-getting-started.html">Getting Started</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-02-data-import-qc.html">Data Import and QC</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-03-complexity-metrics.html">Complexity Metrics</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-04-visualisation.html">Visualisation Gallery</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-05-coexpression.html">Co-expression Analysis</a></li>
  </ul>
</li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-news" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">News</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-news">
<li><h6 class="dropdown-header" data-toc-skip>Releases</h6></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/v1.1.1">Version 1.1.1</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/V1.1.0">Version 1.1.0</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../news/index.html">Changelog</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ThaddeusWu/ScIsoX/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Understanding Transcriptomic Complexity Metrics</h1>
                        <h4 data-toc-skip class="author">Siyuan Wu &amp;
Ulf Schmitz</h4>
            
            <h4 data-toc-skip class="date">2025-08-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ThaddeusWu/ScIsoX/blob/HEAD/vignettes/vignette-03-complexity-metrics.Rmd" class="external-link"><code>vignettes/vignette-03-complexity-metrics.Rmd</code></a></small>
      <div class="d-none name"><code>vignette-03-complexity-metrics.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>ScIsoX quantifies transcriptomic complexity through seven
complementary metrics that capture different aspects of isoform
diversity and heterogeneity. This vignette provides detailed
explanations of each metric and how to interpret them.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ThaddeusWu.github.io/ScIsoX/">ScIsoX</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="the-seven-core-metrics">The Seven Core Metrics<a class="anchor" aria-label="anchor" href="#the-seven-core-metrics"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load example data and create SCHT</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_counts_blood</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">transcript_counts_blood</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">transcript_info</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">sample2stage</span><span class="op">)</span></span>
<span></span>
<span><span class="va">scht_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_scht.html">create_scht</a></span><span class="op">(</span></span>
<span>  gene_counts <span class="op">=</span> <span class="va">gene_counts_blood</span>,</span>
<span>  transcript_counts <span class="op">=</span> <span class="va">transcript_counts_blood</span>,</span>
<span>  transcript_info <span class="op">=</span> <span class="va">transcript_info</span>,</span>
<span>  cell_info <span class="op">=</span> <span class="va">sample2stage</span>,</span>
<span>  qc_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    min_genes_per_cell <span class="op">=</span> <span class="fl">4000</span>,</span>
<span>    max_genes_per_cell <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>    min_cells_expressing <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>    min_expr <span class="op">=</span> <span class="fl">1e-6</span></span>
<span>  <span class="op">)</span>,</span>
<span>  n_hvg <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-03-complexity-metrics_files/figure-html/prepare-data-1.png" width="768"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; === Sparsity Analysis for Current Dataset ===</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 1. Original Transcript Matrix:</span></span>
<span><span class="co">#&gt;   Number of genes: 55,487</span></span>
<span><span class="co">#&gt;   Number of isoforms: 142,238</span></span>
<span><span class="co">#&gt;   Number of cells: 205</span></span>
<span><span class="co">#&gt;   Non-zero elements: 2,457,789</span></span>
<span><span class="co">#&gt;   Zero elements: 26,701,001</span></span>
<span><span class="co">#&gt;   Total elements: 29,158,790</span></span>
<span><span class="co">#&gt;   Sparsity: 91.57%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 2. Filtered Transcript Matrix (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Number of genes: 1,936</span></span>
<span><span class="co">#&gt;   Number of isoforms: 7,327</span></span>
<span><span class="co">#&gt;   Number of cells: 181</span></span>
<span><span class="co">#&gt;   Non-zero elements: 73,057</span></span>
<span><span class="co">#&gt;   Zero elements: 1,253,130</span></span>
<span><span class="co">#&gt;   Total elements: 1,326,187</span></span>
<span><span class="co">#&gt;   Sparsity: 94.49%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 3. SCHT Structure (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Non-zero elements: 73,057</span></span>
<span><span class="co">#&gt;   Zero elements: 154,780</span></span>
<span><span class="co">#&gt;   Total elements: 227,837</span></span>
<span><span class="co">#&gt;   Sparsity: 67.93%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 4. Naive 3D Tensor (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Required dimensions: 1,936 x 21 x 181</span></span>
<span><span class="co">#&gt;   Non-zero elements: 73,057</span></span>
<span><span class="co">#&gt;   Zero elements: 7,285,679</span></span>
<span><span class="co">#&gt;   Total elements: 7,358,736</span></span>
<span><span class="co">#&gt;   Sparsity: 99.01%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 5. Zero Elements Avoided by SCHT:</span></span>
<span><span class="co">#&gt;   vs Original Matrix: 26,546,221</span></span>
<span><span class="co">#&gt;   vs Filtered Matrix: 1,098,350</span></span>
<span><span class="co">#&gt;   vs Naive 3D Tensor: 7,130,899</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 6. Memory Efficiency Summary:</span></span>
<span><span class="co">#&gt;   SCHT uses only 0.78% of original matrix size</span></span>
<span><span class="co">#&gt;   SCHT uses only 17.18% of filtered matrix size</span></span>
<span><span class="co">#&gt;   SCHT uses only 3.10% of naive tensor size</span></span>
<span></span>
<span><span class="co"># Calculate complexity metrics</span></span>
<span><span class="va">tc_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_isoform_complexity_metrics.html">calculate_isoform_complexity_metrics</a></span><span class="op">(</span><span class="va">scht_obj</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
<div class="section level3">
<h3 id="intra-cellular-isoform-diversity">1. Intra-cellular Isoform Diversity<a class="anchor" aria-label="anchor" href="#intra-cellular-isoform-diversity"></a>
</h3>
<p><strong>What it measures</strong>: The tendency of multiple isoforms
from the same gene to be co-expressed within individual cells.</p>
<p><strong>Interpretation</strong>: - <strong>High values</strong>:
Multiple isoforms are frequently co-expressed in the same cells -
<strong>Low values</strong>: Cells typically express only one isoform at
a time</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualise distribution</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">intra_cellular_isoform_diversity</span>,</span>
<span>     main <span class="op">=</span> <span class="st">"Intra-cellular Isoform Diversity Distribution"</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"Diversity Score"</span>,</span>
<span>     col <span class="op">=</span> <span class="st">"lightblue"</span>,</span>
<span>     breaks <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-03-complexity-metrics_files/figure-html/metric-1-1.png" alt="Histogram showing distribution of intra-cellular isoform diversity scores" width="768"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Find genes with highest intra-cellular diversity</span></span>
<span><span class="va">top_intra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">[</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">intra_cellular_isoform_diversity</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"intra_cellular_isoform_diversity"</span>, <span class="st">"n_isoforms"</span><span class="op">)</span></span>
<span><span class="op">]</span>, <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">top_intra</span><span class="op">)</span></span>
<span><span class="co">#&gt;                gene intra_cellular_isoform_diversity n_isoforms</span></span>
<span><span class="co">#&gt; Fsd1l         Fsd1l                        0.9911077          4</span></span>
<span><span class="co">#&gt; Mapk13       Mapk13                        0.9817924          7</span></span>
<span><span class="co">#&gt; Or51ai2     Or51ai2                        0.9584020          3</span></span>
<span><span class="co">#&gt; Ptpro         Ptpro                        0.9530040          3</span></span>
<span><span class="co">#&gt; Hsd11b1     Hsd11b1                        0.9393115          3</span></span>
<span><span class="co">#&gt; Ifitm6       Ifitm6                        0.9380286          2</span></span>
<span><span class="co">#&gt; Ppp1r36dn Ppp1r36dn                        0.9377423          3</span></span>
<span><span class="co">#&gt; Tm4sf1       Tm4sf1                        0.9304544          5</span></span>
<span><span class="co">#&gt; Msantd1     Msantd1                        0.9273881          3</span></span>
<span><span class="co">#&gt; Atp4a         Atp4a                        0.9184838          5</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="inter-cellular-isoform-diversity">2. Inter-cellular Isoform Diversity<a class="anchor" aria-label="anchor" href="#inter-cellular-isoform-diversity"></a>
</h3>
<p><strong>What it measures</strong>: How different isoforms are
distributed across the cell population.</p>
<p><strong>Interpretation</strong>: - <strong>High values</strong>:
Different cells express different isoforms - <strong>Low
values</strong>: All cells express similar isoform patterns</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Compare intra vs inter diversity</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">intra_cellular_isoform_diversity</span>,</span>
<span>     <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">inter_cellular_isoform_diversity</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"Intra-cellular Diversity"</span>,</span>
<span>     ylab <span class="op">=</span> <span class="st">"Inter-cellular Diversity"</span>,</span>
<span>     main <span class="op">=</span> <span class="st">"Intra vs Inter-cellular Diversity"</span>,</span>
<span>     pch <span class="op">=</span> <span class="fl">16</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">0.3</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add correlation</span></span>
<span><span class="va">cor_value</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">intra_cellular_isoform_diversity</span>,</span>
<span>                 <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">inter_cellular_isoform_diversity</span>,</span>
<span>                 use <span class="op">=</span> <span class="st">"complete.obs"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/text.html" class="external-link">text</a></span><span class="op">(</span><span class="fl">0.8</span>, <span class="fl">0.1</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"r ="</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">cor_value</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-03-complexity-metrics_files/figure-html/metric-2-1.png" alt="Scatter plot comparing intra-cellular versus inter-cellular isoform diversity" width="768"></p>
</div>
<div class="section level3">
<h3 id="intra-cell-type-heterogeneity">3. Intra-cell-type Heterogeneity<a class="anchor" aria-label="anchor" href="#intra-cell-type-heterogeneity"></a>
</h3>
<p><strong>What it measures</strong>: Cell-to-cell variation in isoform
usage within the same cell type.</p>
<p><strong>Interpretation</strong>: - <strong>High values</strong>:
Cells of the same type show diverse isoform patterns - <strong>Low
values</strong>: Cells of the same type have consistent isoform
usage</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># This metric requires cell type information</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">inherits</a></span><span class="op">(</span><span class="va">scht_obj</span>, <span class="st">"IntegratedSCHT"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Visualise by classification</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span>, </span>
<span>         <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">intra_cell_type_heterogeneity_class</span>, </span>
<span>             fill <span class="op">=</span> <span class="va">intra_cell_type_heterogeneity_class</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Distribution of Intra-cell-type Heterogeneity Classes"</span>,</span>
<span>         x <span class="op">=</span> <span class="st">"Heterogeneity Class"</span>, y <span class="op">=</span> <span class="st">"Number of Genes"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="vignette-03-complexity-metrics_files/figure-html/metric-3-1.png" alt="Bar chart showing distribution of intra-cell-type heterogeneity classes" width="768"></p>
</div>
<div class="section level3">
<h3 id="inter-cell-type-specificity">4. Inter-cell-type Specificity<a class="anchor" aria-label="anchor" href="#inter-cell-type-specificity"></a>
</h3>
<p><strong>What it measures</strong>: How specific isoform usage
patterns are to particular cell types.</p>
<p><strong>Interpretation</strong>: - <strong>High values</strong>:
Strong cell type-specific isoform usage - <strong>Low values</strong>:
Similar isoform patterns across cell types</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find cell type-specific genes</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="st">"inter_cell_type_specificity"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">ct_specific</span> <span class="op">&lt;-</span> <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">[</span></span>
<span>    <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">inter_cell_type_specificity_class</span> <span class="op">==</span> <span class="st">"Cell-Type-Specific Isoform Expression"</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"inter_cell_type_specificity"</span><span class="op">)</span></span>
<span>  <span class="op">]</span></span>
<span>  </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Found"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">ct_specific</span><span class="op">)</span>, <span class="st">"genes with high cell type specificity"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ct_specific</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; [1] "Found 636 genes with high cell type specificity"</span></span>
<span><span class="co">#&gt;              gene inter_cell_type_specificity</span></span>
<span><span class="co">#&gt; Cd177       Cd177                           1</span></span>
<span><span class="co">#&gt; S100a8     S100a8                           1</span></span>
<span><span class="co">#&gt; S100a9     S100a9                           1</span></span>
<span><span class="co">#&gt; Shtn1       Shtn1                           1</span></span>
<span><span class="co">#&gt; Ceacam10 Ceacam10                           1</span></span>
<span><span class="co">#&gt; Oas3         Oas3                           1</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="intra-cell-type-heterogeneity-variability">5. Intra-cell-type Heterogeneity Variability<a class="anchor" aria-label="anchor" href="#intra-cell-type-heterogeneity-variability"></a>
</h3>
<p><strong>What it measures</strong>: How much the cell-to-cell
heterogeneity varies across different cell types.</p>
<p><strong>Interpretation</strong>: - <strong>High values</strong>: Some
cell types are more heterogeneous than others - <strong>Low
values</strong>: All cell types show similar levels of heterogeneity</p>
</div>
<div class="section level3">
<h3 id="inter-cell-type-difference-variability">6. Inter-cell-type Difference Variability<a class="anchor" aria-label="anchor" href="#inter-cell-type-difference-variability"></a>
</h3>
<p><strong>What it measures</strong>: Identifies pairs of cell types
with the most distinctive isoform usage patterns.</p>
<p><strong>Interpretation</strong>: - <strong>High values</strong>:
Large differences between some cell type pairs - <strong>Low
values</strong>: Uniform differences across all cell type pairs</p>
</div>
<div class="section level3">
<h3 id="cell-type-specific-co-expression-variability">7. Cell-type-specific Co-expression Variability<a class="anchor" aria-label="anchor" href="#cell-type-specific-co-expression-variability"></a>
</h3>
<p><strong>What it measures</strong>: How isoform co-expression patterns
vary across cell types.</p>
<p><strong>Interpretation</strong>: - <strong>High values</strong>:
Co-expression relationships change between cell types - <strong>Low
values</strong>: Consistent co-expression across cell types</p>
</div>
</div>
<div class="section level2">
<h2 id="interpreting-metric-classifications">Interpreting Metric Classifications<a class="anchor" aria-label="anchor" href="#interpreting-metric-classifications"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Summary of classifications</span></span>
<span><span class="va">classification_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"_class$"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  <span class="kw">function</span><span class="op">(</span><span class="va">col</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">[[</span><span class="va">col</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">classification_summary</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"_class$"</span>, <span class="st">""</span>, </span>
<span>                                     <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"_class$"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">classification_summary</span><span class="op">)</span></span>
<span><span class="co">#&gt; $intra_cellular_isoform_diversity</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Strong Isoform Co-expression   Weak Isoform Co-expression </span></span>
<span><span class="co">#&gt;                          704                         1232 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $inter_cellular_isoform_diversity</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; High Isoform Diversity  Low Isoform Diversity </span></span>
<span><span class="co">#&gt;                    595                   1341 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $intra_cell_type_heterogeneity</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; High Cellular Heterogeneity  Low Cellular Heterogeneity </span></span>
<span><span class="co">#&gt;                         819                         954 </span></span>
<span><span class="co">#&gt;                Unclassified </span></span>
<span><span class="co">#&gt;                         163 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $inter_cell_type_specificity</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Cell-Type-Independent Isoform Expression </span></span>
<span><span class="co">#&gt;                                     1300 </span></span>
<span><span class="co">#&gt;    Cell-Type-Specific Isoform Expression </span></span>
<span><span class="co">#&gt;                                      636 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $intra_cell_type_heterogeneity_variability</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Consistent Heterogeneity Across Cell Types </span></span>
<span><span class="co">#&gt;                                        892 </span></span>
<span><span class="co">#&gt;                Insufficient Cell Type Data </span></span>
<span><span class="co">#&gt;                                        492 </span></span>
<span><span class="co">#&gt;   Variable Heterogeneity Across Cell Types </span></span>
<span><span class="co">#&gt;                                        552 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $inter_cell_type_difference_variability</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  High Cell-Type Distinctions Insufficient Difference Data </span></span>
<span><span class="co">#&gt;                          943                          137 </span></span>
<span><span class="co">#&gt;   Low Cell-Type Distinctions </span></span>
<span><span class="co">#&gt;                          856 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $cell_type_coexpression_variability</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Cell-Type-Adaptive Co-expression Cell-Type-Consistent Co-expression </span></span>
<span><span class="co">#&gt;                                596                               1156 </span></span>
<span><span class="co">#&gt;                  Insufficient Data </span></span>
<span><span class="co">#&gt;                                184</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="finding-genes-with-specific-patterns">Finding Genes with Specific Patterns<a class="anchor" aria-label="anchor" href="#finding-genes-with-specific-patterns"></a>
</h2>
<div class="section level3">
<h3 id="example-1-highly-complex-genes">Example 1: Highly Complex Genes<a class="anchor" aria-label="anchor" href="#example-1-highly-complex-genes"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Genes with high complexity in multiple dimensions</span></span>
<span><span class="va">complex_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find_complexity_pattern.html">find_complexity_pattern</a></span><span class="op">(</span></span>
<span>  <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span>,</span>
<span>  pattern <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    intra_cellular_isoform_diversity_class <span class="op">=</span> <span class="st">"Strong Isoform Co-expression"</span>,</span>
<span>    inter_cellular_isoform_diversity_class <span class="op">=</span> <span class="st">"High Isoform Diversity"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Found"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">complex_genes</span><span class="op">)</span>, <span class="st">"highly complex genes"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Found 20 highly complex genes"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">complex_genes</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Cd177"   "S100a8"  "S100a9"  "Il7"     "Nup62cl" "Crb3"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example-2-simple-but-cell-type-specific">Example 2: Simple but Cell Type-Specific<a class="anchor" aria-label="anchor" href="#example-2-simple-but-cell-type-specific"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Genes with low diversity but high cell type specificity</span></span>
<span><span class="va">specific_simple</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find_complexity_pattern.html">find_complexity_pattern</a></span><span class="op">(</span></span>
<span>  <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span>,</span>
<span>  pattern <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    intra_cellular_isoform_diversity_class <span class="op">=</span> <span class="st">"Weak Isoform Co-expression"</span>,</span>
<span>    inter_cell_type_specificity_class <span class="op">=</span> <span class="st">"Cell-Type-Specific Isoform Expression"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">specific_simple</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Found"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">specific_simple</span><span class="op">)</span>, <span class="st">"simple but specific genes"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; [1] "Found 20 simple but specific genes"</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualising-metric-relationships">Visualising Metric Relationships<a class="anchor" aria-label="anchor" href="#visualising-metric-relationships"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define all seven core metrics</span></span>
<span><span class="va">metric_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"intra_cellular_isoform_diversity"</span>,</span>
<span>  <span class="st">"inter_cellular_isoform_diversity"</span>,</span>
<span>  <span class="st">"inter_cell_type_specificity"</span>,</span>
<span>  <span class="st">"intra_cell_type_heterogeneity"</span>,</span>
<span>  <span class="st">"intra_cell_type_heterogeneity_variability"</span>,</span>
<span>  <span class="st">"inter_cell_type_difference_variability"</span>,</span>
<span>  <span class="st">"cell_type_coexpression_variability"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to available metrics in the dataset</span></span>
<span><span class="va">available_metrics</span> <span class="op">&lt;-</span> <span class="va">metric_cols</span><span class="op">[</span><span class="va">metric_cols</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">metric_data</span> <span class="op">&lt;-</span> <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">[</span>, <span class="va">available_metrics</span><span class="op">]</span></span>
<span><span class="va">metric_data</span> <span class="op">&lt;-</span> <span class="va">metric_data</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">metric_data</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Standardise the data for PCA</span></span>
<span><span class="va">metric_scaled</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="va">metric_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform PCA analysis</span></span>
<span><span class="va">pca_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></span><span class="op">(</span><span class="va">metric_scaled</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create data for biplot</span></span>
<span><span class="va">pca_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">pca_result</span><span class="op">$</span><span class="va">x</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">pca_scores</span><span class="op">$</span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">gene</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">[</span>, <span class="va">available_metrics</span><span class="op">]</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Extract loadings (metric contributions)</span></span>
<span><span class="va">pca_loadings</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">pca_result</span><span class="op">$</span><span class="va">rotation</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">pca_loadings</span><span class="op">$</span><span class="va">metric</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pca_loadings</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Scale loadings for better visualisation</span></span>
<span><span class="va">scale_factor</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span><span class="va">pca_loadings</span><span class="op">$</span><span class="va">PC1</span> <span class="op">&lt;-</span> <span class="va">pca_loadings</span><span class="op">$</span><span class="va">PC1</span> <span class="op">*</span> <span class="va">scale_factor</span></span>
<span><span class="va">pca_loadings</span><span class="op">$</span><span class="va">PC2</span> <span class="op">&lt;-</span> <span class="va">pca_loadings</span><span class="op">$</span><span class="va">PC2</span> <span class="op">*</span> <span class="va">scale_factor</span></span>
<span></span>
<span><span class="co"># Create PCA biplot</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggrepel.slowkow.com/" class="external-link">ggrepel</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># Gene points (in grey)</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">pca_scores</span>, </span>
<span>             <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span><span class="op">)</span>, </span>
<span>             alpha <span class="op">=</span> <span class="fl">0.5</span>, colour <span class="op">=</span> <span class="st">"grey50"</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># Metric vectors (red arrows)</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_segment.html" class="external-link">geom_segment</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">pca_loadings</span>,</span>
<span>               <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fl">0</span>, y <span class="op">=</span> <span class="fl">0</span>, xend <span class="op">=</span> <span class="va">PC1</span>, yend <span class="op">=</span> <span class="va">PC2</span><span class="op">)</span>,</span>
<span>               arrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/arrow.html" class="external-link">arrow</a></span><span class="op">(</span>length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="st">"cm"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>               colour <span class="op">=</span> <span class="st">"red"</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># Metric labels</span></span>
<span>  <span class="fu"><a href="https://ggrepel.slowkow.com/reference/geom_text_repel.html" class="external-link">geom_text_repel</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">pca_loadings</span>,</span>
<span>                  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span>, label <span class="op">=</span> <span class="va">metric</span><span class="op">)</span>,</span>
<span>                  colour <span class="op">=</span> <span class="st">"red"</span>, fontface <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"PCA Biplot of Complexity Metrics"</span>,</span>
<span>       subtitle <span class="op">=</span> <span class="st">"Red arrows show metric relationships; genes shown as grey points"</span>,</span>
<span>       x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"PC1 ("</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">pca_result</span><span class="op">)</span><span class="op">$</span><span class="va">importance</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">1</span><span class="op">]</span><span class="op">*</span><span class="fl">100</span>, <span class="fl">1</span><span class="op">)</span>, <span class="st">"%)"</span><span class="op">)</span>,</span>
<span>       y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"PC2 ("</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">pca_result</span><span class="op">)</span><span class="op">$</span><span class="va">importance</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">]</span><span class="op">*</span><span class="fl">100</span>, <span class="fl">1</span><span class="op">)</span>, <span class="st">"%)"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>        plot.subtitle <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_equal</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-03-complexity-metrics_files/figure-html/relationships-1.png" alt="PCA biplot showing relationships between complexity metrics" width="960"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Print variance explained</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Variance explained by principal components:\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Variance explained by principal components:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">pca_result</span><span class="op">)</span><span class="op">$</span><span class="va">importance</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;                             PC1       PC2       PC3</span></span>
<span><span class="co">#&gt; Standard deviation     2.059705 0.9829898 0.7287917</span></span>
<span><span class="co">#&gt; Proportion of Variance 0.606060 0.1380400 0.0758800</span></span>
<span><span class="co">#&gt; Cumulative Proportion  0.606060 0.7440900 0.8199700</span></span></code></pre></div>
<div class="section level3">
<h3 id="interpretation">Interpretation:<a class="anchor" aria-label="anchor" href="#interpretation"></a>
</h3>
<ol style="list-style-type: decimal">
<li>
<strong>Arrows pointing in similar directions</strong>: Metrics that
tend to increase together</li>
<li>
<strong>Arrows pointing in opposite directions</strong>: Metrics
that are inversely related</li>
<li>
<strong>Arrow length</strong>: Indicates how well the metric is
represented in the first two PCs</li>
<li>
<strong>Gene distribution</strong>: Shows how genes cluster based on
their complexity profiles</li>
</ol>
<p><em>Note: The specific relationships between metrics will vary
depending on your dataset and biological context.</em></p>
</div>
</div>
<div class="section level2">
<h2 id="case-studies">Case Studies<a class="anchor" aria-label="anchor" href="#case-studies"></a>
</h2>
<div class="section level3">
<h3 id="case-study-1-constitutive-vs-alternative-splicing">Case Study 1: Constitutive vs Alternative Splicing<a class="anchor" aria-label="anchor" href="#case-study-1-constitutive-vs-alternative-splicing"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Compare genes with different numbers of isoforms</span></span>
<span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">isoform_category</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cut.html" class="external-link">cut</a></span><span class="op">(</span></span>
<span>  <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">n_isoforms</span>,</span>
<span>  breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">5</span>, <span class="fl">10</span>, <span class="cn">Inf</span><span class="op">)</span>,</span>
<span>  labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"2-3"</span>, <span class="st">"4-5"</span>, <span class="st">"6-10"</span>, <span class="st">"&gt;10"</span><span class="op">)</span>,</span>
<span>  include.lowest <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># How does complexity relate to number of isoforms?</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span>, </span>
<span>       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">isoform_category</span>, y <span class="op">=</span> <span class="va">intra_cellular_isoform_diversity</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"lightblue"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Complexity vs Number of Isoforms"</span>,</span>
<span>       x <span class="op">=</span> <span class="st">"Number of Isoforms"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"Intra-cellular Diversity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-03-complexity-metrics_files/figure-html/case-study-1-1.png" alt="Boxplot comparing complexity across different numbers of isoforms" width="768"></p>
</div>
<div class="section level3">
<h3 id="case-study-2-identifying-regulatory-genes">Case Study 2: Identifying Regulatory Genes<a class="anchor" aria-label="anchor" href="#case-study-2-identifying-regulatory-genes"></a>
</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Genes with variable complexity patterns might be regulatory</span></span>
<span><span class="co"># High inter-cellular diversity but low intra-cellular diversity</span></span>
<span><span class="co"># suggests cell-state dependent isoform switching</span></span>
<span></span>
<span><span class="va">potential_regulatory</span> <span class="op">&lt;-</span> <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">[</span></span>
<span>  <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">inter_cellular_isoform_diversity</span> <span class="op">&gt;</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">inter_cellular_isoform_diversity</span>, <span class="fl">0.75</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">&amp;</span></span>
<span>  <span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">intra_cellular_isoform_diversity</span> <span class="op">&lt;</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span><span class="op">$</span><span class="va">intra_cellular_isoform_diversity</span>, <span class="fl">0.25</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Identified"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">potential_regulatory</span><span class="op">)</span>, <span class="st">"potential regulatory genes"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Identified 57 potential regulatory genes"</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">potential_regulatory</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">potential_regulatory</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"n_isoforms"</span>,</span>
<span>                                <span class="st">"intra_cellular_isoform_diversity"</span>,</span>
<span>                                <span class="st">"inter_cellular_isoform_diversity"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt;                        gene n_isoforms intra_cellular_isoform_diversity</span></span>
<span><span class="co">#&gt; B3galt1             B3galt1          2                                0</span></span>
<span><span class="co">#&gt; Cxxc4                 Cxxc4          2                                0</span></span>
<span><span class="co">#&gt; Tac1                   Tac1          2                                0</span></span>
<span><span class="co">#&gt; Dnah7a               Dnah7a          2                                0</span></span>
<span><span class="co">#&gt; 1700019L13Rik 1700019L13Rik          2                                0</span></span>
<span><span class="co">#&gt; Fbxl16               Fbxl16          2                                0</span></span>
<span><span class="co">#&gt;               inter_cellular_isoform_diversity</span></span>
<span><span class="co">#&gt; B3galt1                              0.9754361</span></span>
<span><span class="co">#&gt; Cxxc4                                0.9466857</span></span>
<span><span class="co">#&gt; Tac1                                 0.9333993</span></span>
<span><span class="co">#&gt; Dnah7a                               0.9651509</span></span>
<span><span class="co">#&gt; 1700019L13Rik                        0.8662512</span></span>
<span><span class="co">#&gt; Fbxl16                               0.9504299</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="exporting-results">Exporting Results<a class="anchor" aria-label="anchor" href="#exporting-results"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Export metrics for further analysis</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">$</span><span class="va">metrics</span>, </span>
<span>          file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"complexity_metrics.csv"</span><span class="op">)</span>,</span>
<span>          row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create summary report</span></span>
<span><span class="va">summary_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">tc_results</span><span class="op">)</span></span>
<span><span class="co">#&gt; Isoform Complexity Analysis Summary:</span></span>
<span><span class="co">#&gt; Total genes analysed: 1936</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; NA Value Statistics (representing biologically meaningful cases):</span></span>
<span><span class="co">#&gt;   - Intra-cellular Isoform Diversity   :    0 genes (0.0%) - Occurs when genes have only one expressed isoform, indicating lack of alternative splicing</span></span>
<span><span class="co">#&gt;   - Inter-cellular Isoform Diversity   :    0 genes (0.0%) - Occurs when only one isoform is expressed across the entire cell population</span></span>
<span><span class="co">#&gt;   - Intra-cell-type Heterogeneity      :  163 genes (8.4%) - Occurs when a cell type has insufficient cells expressing the gene (&lt; 3 cells)</span></span>
<span><span class="co">#&gt;   - Inter-cell-type Specificity        :    0 genes (0.0%) - Occurs for genes expressed in only a single cell type (high cell type-specificity)</span></span>
<span><span class="co">#&gt;   - Intra-cell-type Heterogeneity Variability:  492 genes (25.4%) - Occurs when a gene is expressed in fewer than 2 cell types with sufficient data</span></span>
<span><span class="co">#&gt;   - Inter-cell-type Difference Variability:  137 genes (7.1%) - Occurs when there are insufficient pairwise differences between cell types to calculate variability</span></span>
<span><span class="co">#&gt;   - Cell-type-specific Co-expression Variability:  184 genes (9.5%) - Occurs when a gene lacks sufficient cell type-specific expression data to compare co-expression patterns</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Classification Distribution Across Complexity Dimensions:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Intra-cellular Isoform Diversity:</span></span>
<span><span class="co">#&gt;   - Strong Isoform Co-expression            :  704 genes (36.4%)</span></span>
<span><span class="co">#&gt;   - Weak Isoform Co-expression              : 1232 genes (63.6%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Inter-cellular Isoform Diversity:</span></span>
<span><span class="co">#&gt;   - High Isoform Diversity                  :  595 genes (30.7%)</span></span>
<span><span class="co">#&gt;   - Low Isoform Diversity                   : 1341 genes (69.3%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Intra-cell-type Heterogeneity:</span></span>
<span><span class="co">#&gt;   - High Cellular Heterogeneity             :  819 genes (42.3%)</span></span>
<span><span class="co">#&gt;   - Low Cellular Heterogeneity              :  954 genes (49.3%)</span></span>
<span><span class="co">#&gt;   - Unclassified                            :  163 genes (8.4%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Inter-cell-type Specificity:</span></span>
<span><span class="co">#&gt;   - Cell-Type-Independent Isoform Expression: 1300 genes (67.1%)</span></span>
<span><span class="co">#&gt;   - Cell-Type-Specific Isoform Expression   :  636 genes (32.9%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Intra-cell-type Heterogeneity Variability:</span></span>
<span><span class="co">#&gt;   - Consistent Heterogeneity Across Cell Types:  892 genes (46.1%)</span></span>
<span><span class="co">#&gt;   - Insufficient Cell Type Data             :  492 genes (25.4%)</span></span>
<span><span class="co">#&gt;   - Variable Heterogeneity Across Cell Types:  552 genes (28.5%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Inter-cell-type Difference Variability:</span></span>
<span><span class="co">#&gt;   - High Cell-Type Distinctions             :  943 genes (48.7%)</span></span>
<span><span class="co">#&gt;   - Insufficient Difference Data            :  137 genes (7.1%)</span></span>
<span><span class="co">#&gt;   - Low Cell-Type Distinctions              :  856 genes (44.2%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Cell-type-specific Co-expression Variability:</span></span>
<span><span class="co">#&gt;   - Cell-Type-Adaptive Co-expression        :  596 genes (30.8%)</span></span>
<span><span class="co">#&gt;   - Cell-Type-Consistent Co-expression      : 1156 genes (59.7%)</span></span>
<span><span class="co">#&gt;   - Insufficient Data                       :  184 genes (9.5%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Traditional Complexity Categories:</span></span>
<span><span class="co">#&gt;   - High Diversity + High Specificity       :  319 genes (16.5%)</span></span>
<span><span class="co">#&gt;   - High Diversity + Low Specificity        :  276 genes (14.3%)</span></span>
<span><span class="co">#&gt;   - Low Diversity + High Specificity        :  317 genes (16.4%)</span></span>
<span><span class="co">#&gt;   - Low Diversity + Low Specificity         : 1024 genes (52.9%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Cell-Type-Specific Genes:</span></span>
<span><span class="co">#&gt;   - Single cell-type genes                  :   23 genes (1.2%)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Core Metrics Statistics:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; |Metric                                       |  Mean| Median|    SD|  Min|   Max| Threshold| NA_Percent|</span></span>
<span><span class="co">#&gt; |:--------------------------------------------|-----:|------:|-----:|----:|-----:|---------:|----------:|</span></span>
<span><span class="co">#&gt; |Intra-cellular Isoform Diversity             | 0.351|  0.321| 0.270| 0.00| 0.991|     0.456|      0.000|</span></span>
<span><span class="co">#&gt; |Inter-cellular Isoform Diversity             | 0.591|  0.630| 0.257| 0.01| 1.000|     0.758|      0.000|</span></span>
<span><span class="co">#&gt; |Intra-cell-type Heterogeneity                | 0.363|  0.367| 0.230| 0.00| 1.000|     0.389|      8.419|</span></span>
<span><span class="co">#&gt; |Inter-cell-type Specificity                  | 0.401|  0.382| 0.230| 0.00| 1.000|     0.497|      0.000|</span></span>
<span><span class="co">#&gt; |Intra-cell-type Heterogeneity Variability    | 0.770|  0.620| 0.625| 0.00| 2.646|     0.832|     25.413|</span></span>
<span><span class="co">#&gt; |Inter-cell-type Difference Variability       | 0.646|  0.567| 0.342| 0.00| 1.620|     0.552|      7.076|</span></span>
<span><span class="co">#&gt; |Cell-type-specific Co-expression Variability | 1.032|  0.972| 0.702| 0.00| 2.646|     1.389|      9.504|</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Performance metrics:</span></span>
<span><span class="co">#&gt;   Processing time: 82.89 seconds (1.38 minutes)</span></span>
<span><span class="co">#&gt;   Memory utilised: 1011.84 MB</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="advanced-interpretation-tips">Advanced Interpretation Tips<a class="anchor" aria-label="anchor" href="#advanced-interpretation-tips"></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<strong>Look for outliers</strong>: Genes with extreme values often
have interesting biology</li>
<li>
<strong>Consider metric combinations</strong>: Single metrics tell
only part of the story</li>
<li>
<strong>Relate to gene function</strong>: Different gene categories
show characteristic patterns</li>
<li>
<strong>Account for technical factors</strong>: Low expression can
affect metric reliability</li>
</ol>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.3 (2025-02-28)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Sequoia 15.5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Australia/Brisbane</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] grid      stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ggrepel_0.9.6         ComplexHeatmap_2.22.0 ggplot2_3.5.2        </span></span>
<span><span class="co">#&gt; [4] ScIsoX_1.1.1         </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] bitops_1.0-9                gridExtra_2.3              </span></span>
<span><span class="co">#&gt;   [3] rlang_1.1.6                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [5] clue_0.3-66                 GetoptLong_1.0.5           </span></span>
<span><span class="co">#&gt;   [7] ggridges_0.5.6              matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt;   [9] compiler_4.4.3              png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [11] systemfonts_1.2.1           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;  [13] pkgconfig_2.0.3             shape_1.4.6.1              </span></span>
<span><span class="co">#&gt;  [15] crayon_1.5.3                fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [17] XVector_0.46.0              labeling_0.4.3             </span></span>
<span><span class="co">#&gt;  [19] promises_1.3.3              Rsamtools_2.22.0           </span></span>
<span><span class="co">#&gt;  [21] rmarkdown_2.29              UCSC.utils_1.2.0           </span></span>
<span><span class="co">#&gt;  [23] ragg_1.4.0                  purrr_1.1.0                </span></span>
<span><span class="co">#&gt;  [25] xfun_0.52                   zlibbioc_1.52.0            </span></span>
<span><span class="co">#&gt;  [27] cachem_1.1.0                GenomeInfoDb_1.42.3        </span></span>
<span><span class="co">#&gt;  [29] jsonlite_2.0.0              progress_1.2.3             </span></span>
<span><span class="co">#&gt;  [31] later_1.4.2                 DelayedArray_0.32.0        </span></span>
<span><span class="co">#&gt;  [33] BiocParallel_1.40.0         parallel_4.4.3             </span></span>
<span><span class="co">#&gt;  [35] prettyunits_1.2.0           cluster_2.1.8.1            </span></span>
<span><span class="co">#&gt;  [37] R6_2.6.1                    bslib_0.9.0                </span></span>
<span><span class="co">#&gt;  [39] RColorBrewer_1.1-3          rtracklayer_1.66.0         </span></span>
<span><span class="co">#&gt;  [41] car_3.1-3                   GenomicRanges_1.58.0       </span></span>
<span><span class="co">#&gt;  [43] jquerylib_0.1.4             diptest_0.77-1             </span></span>
<span><span class="co">#&gt;  [45] Rcpp_1.1.0                  SummarizedExperiment_1.36.0</span></span>
<span><span class="co">#&gt;  [47] iterators_1.0.14            knitr_1.50                 </span></span>
<span><span class="co">#&gt;  [49] IRanges_2.40.1              httpuv_1.6.16              </span></span>
<span><span class="co">#&gt;  [51] Matrix_1.7-3                tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt;  [53] viridis_0.6.5               rstudioapi_0.17.1          </span></span>
<span><span class="co">#&gt;  [55] abind_1.4-8                 yaml_2.3.10                </span></span>
<span><span class="co">#&gt;  [57] miniUI_0.1.1.1              doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [59] codetools_0.2-20            curl_6.4.0                 </span></span>
<span><span class="co">#&gt;  [61] lattice_0.22-6              tibble_3.3.0               </span></span>
<span><span class="co">#&gt;  [63] plyr_1.8.9                  withr_3.0.2                </span></span>
<span><span class="co">#&gt;  [65] shiny_1.11.1                Biobase_2.66.0             </span></span>
<span><span class="co">#&gt;  [67] evaluate_1.0.4              moments_0.14.1             </span></span>
<span><span class="co">#&gt;  [69] desc_1.4.3                  circlize_0.4.16            </span></span>
<span><span class="co">#&gt;  [71] mclust_6.1.1                Biostrings_2.74.1          </span></span>
<span><span class="co">#&gt;  [73] pillar_1.11.0               MatrixGenerics_1.18.1      </span></span>
<span><span class="co">#&gt;  [75] carData_3.0-5               DT_0.33                    </span></span>
<span><span class="co">#&gt;  [77] foreach_1.5.2               stats4_4.4.3               </span></span>
<span><span class="co">#&gt;  [79] plotly_4.11.0               generics_0.1.4             </span></span>
<span><span class="co">#&gt;  [81] RCurl_1.98-1.16             S4Vectors_0.44.0           </span></span>
<span><span class="co">#&gt;  [83] hms_1.1.3                   scales_1.4.0               </span></span>
<span><span class="co">#&gt;  [85] xtable_1.8-4                glue_1.8.0                 </span></span>
<span><span class="co">#&gt;  [87] lazyeval_0.2.2              tools_4.4.3                </span></span>
<span><span class="co">#&gt;  [89] BiocIO_1.16.0               data.table_1.17.8          </span></span>
<span><span class="co">#&gt;  [91] ggradar_0.2                 GenomicAlignments_1.42.0   </span></span>
<span><span class="co">#&gt;  [93] fs_1.6.6                    XML_3.99-0.18              </span></span>
<span><span class="co">#&gt;  [95] cowplot_1.2.0               tidyr_1.3.1                </span></span>
<span><span class="co">#&gt;  [97] colorspace_2.1-1            patchwork_1.3.0            </span></span>
<span><span class="co">#&gt;  [99] GenomeInfoDbData_1.2.13     restfulr_0.0.15            </span></span>
<span><span class="co">#&gt; [101] Formula_1.2-5               cli_3.6.5                  </span></span>
<span><span class="co">#&gt; [103] textshaping_1.0.0           S4Arrays_1.6.0             </span></span>
<span><span class="co">#&gt; [105] viridisLite_0.4.2           dplyr_1.1.4                </span></span>
<span><span class="co">#&gt; [107] gtable_0.3.6                sass_0.4.10                </span></span>
<span><span class="co">#&gt; [109] digest_0.6.37               BiocGenerics_0.52.0        </span></span>
<span><span class="co">#&gt; [111] SparseArray_1.6.2           rjson_0.2.23               </span></span>
<span><span class="co">#&gt; [113] htmlwidgets_1.6.4           farver_2.1.2               </span></span>
<span><span class="co">#&gt; [115] htmltools_0.5.8.1           pkgdown_2.1.3              </span></span>
<span><span class="co">#&gt; [117] lifecycle_1.0.4             httr_1.4.7                 </span></span>
<span><span class="co">#&gt; [119] mime_0.13                   GlobalOptions_0.1.2        </span></span>
<span><span class="co">#&gt; [121] ggExtra_0.10.1              MASS_7.3-65</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Siyuan Wu, Ulf Schmitz<br> Site built with <a href="https://pkgdown.r-lib.org" class="external-link">pkgdown</a> 2.1.1</p>
</div>

<div class="pkgdown-footer-right">
  <p><strong>Package License</strong>: <a href="https://opensource.org/licenses/MIT" class="external-link">MIT</a><br><br><strong>Example Data Attribution</strong>:<br> The example datasets included in this package are derived from:<br> Wang et al. (2022). Single-cell multiomics defines tolerogenic extrathymic Aire-expressing populations. <em>Science Advances</em> 8(1):eabg5369<br> Licensed under <a href="https://creativecommons.org/licenses/by/4.0/" class="external-link">CC BY 4.0</a><br> We gratefully acknowledge the authors for making their data publicly available.</p>
</div>

    </footer>
</div>





  </body>
</html>
