<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Co-expression Analysis and Isoform Switching • ScIsoX</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Co-expression Analysis and Isoform Switching">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-#1a5490" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ScIsoX</a>

    <small class="nav-text text-success me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.1.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/vignette-01-getting-started.html">Getting Started</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-02-data-import-qc.html">Data Import and QC</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-03-complexity-metrics.html">Complexity Metrics</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-04-visualisation.html">Visualisation Gallery</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-05-coexpression.html">Co-expression Analysis</a></li>
  </ul>
</li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-news" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">News</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-news">
<li><h6 class="dropdown-header" data-toc-skip>Releases</h6></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/v1.1.3">Version 1.1.3</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/v1.1.2">Version 1.1.2</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/v1.1.1">Version 1.1.1</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/V1.1.0">Version 1.1.0</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../news/index.html">Changelog</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ThaddeusWu/ScIsoX/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Co-expression Analysis and Isoform Switching</h1>
                        <h4 data-toc-skip class="author">Siyuan Wu &amp;
Ulf Schmitz</h4>
            
            <h4 data-toc-skip class="date">2025-09-28</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ThaddeusWu/ScIsoX/blob/HEAD/vignettes/vignette-05-coexpression.Rmd" class="external-link"><code>vignettes/vignette-05-coexpression.Rmd</code></a></small>
      <div class="d-none name"><code>vignette-05-coexpression.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette demonstrates ScIsoX’s comprehensive co-expression
analysis capabilities through the interactive Shiny application, which
provides an intuitive interface for exploring isoform relationships,
switching patterns, and cell type-specific co-expression.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ThaddeusWu.github.io/ScIsoX/">ScIsoX</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load example data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_counts_blood</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">transcript_counts_blood</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">transcript_info</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">sample2stage</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create IntegratedSCHT for cell type-specific analysis</span></span>
<span><span class="va">integrated_scht</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_scht.html">create_scht</a></span><span class="op">(</span></span>
<span>  gene_counts <span class="op">=</span> <span class="va">gene_counts_blood</span>,</span>
<span>  transcript_counts <span class="op">=</span> <span class="va">transcript_counts_blood</span>,</span>
<span>  transcript_info <span class="op">=</span> <span class="va">transcript_info</span>,</span>
<span>  cell_info <span class="op">=</span> <span class="va">sample2stage</span>,</span>
<span>  qc_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    min_genes_per_cell <span class="op">=</span> <span class="fl">4000</span>,</span>
<span>    max_genes_per_cell <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>    min_cells_expressing <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>    min_expr <span class="op">=</span> <span class="fl">1e-6</span></span>
<span>  <span class="op">)</span>,</span>
<span>  n_hvg <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  require_cell_type <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette-05-coexpression_files/figure-html/load-package-1.png" width="768"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; === Sparsity Analysis for Current Dataset ===</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 1. Original Transcript Matrix:</span></span>
<span><span class="co">#&gt;   Number of genes: 55,487</span></span>
<span><span class="co">#&gt;   Number of isoforms: 142,238</span></span>
<span><span class="co">#&gt;   Number of cells: 205</span></span>
<span><span class="co">#&gt;   Non-zero elements: 2,457,789</span></span>
<span><span class="co">#&gt;   Zero elements: 26,701,001</span></span>
<span><span class="co">#&gt;   Total elements: 29,158,790</span></span>
<span><span class="co">#&gt;   Sparsity: 91.57%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 2. Filtered Transcript Matrix (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Number of genes: 1,936</span></span>
<span><span class="co">#&gt;   Number of isoforms: 7,327</span></span>
<span><span class="co">#&gt;   Number of cells: 181</span></span>
<span><span class="co">#&gt;   Non-zero elements: 73,057</span></span>
<span><span class="co">#&gt;   Zero elements: 1,253,130</span></span>
<span><span class="co">#&gt;   Total elements: 1,326,187</span></span>
<span><span class="co">#&gt;   Sparsity: 94.49%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 3. SCHT Structure (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Non-zero elements: 73,057</span></span>
<span><span class="co">#&gt;   Zero elements: 154,780</span></span>
<span><span class="co">#&gt;   Total elements: 227,837</span></span>
<span><span class="co">#&gt;   Sparsity: 67.93%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 4. Naive 3D Tensor (Post-QC HVG):</span></span>
<span><span class="co">#&gt;   Required dimensions: 1,936 x 21 x 181</span></span>
<span><span class="co">#&gt;   Non-zero elements: 73,057</span></span>
<span><span class="co">#&gt;   Zero elements: 7,285,679</span></span>
<span><span class="co">#&gt;   Total elements: 7,358,736</span></span>
<span><span class="co">#&gt;   Sparsity: 99.01%</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 5. Zero Elements Avoided by SCHT:</span></span>
<span><span class="co">#&gt;   vs Original Matrix: 26,546,221</span></span>
<span><span class="co">#&gt;   vs Filtered Matrix: 1,098,350</span></span>
<span><span class="co">#&gt;   vs Naive 3D Tensor: 7,130,899</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 6. Memory Efficiency Summary:</span></span>
<span><span class="co">#&gt;   SCHT uses only 0.78% of original matrix size</span></span>
<span><span class="co">#&gt;   SCHT uses only 17.18% of filtered matrix size</span></span>
<span><span class="co">#&gt;   SCHT uses only 3.10% of naive tensor size</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="interactive-co-expression-explorer">Interactive Co-expression Explorer<a class="anchor" aria-label="anchor" href="#interactive-co-expression-explorer"></a>
</h2>
<div class="section level3">
<h3 id="launching-the-shiny-app">Launching the Shiny App<a class="anchor" aria-label="anchor" href="#launching-the-shiny-app"></a>
</h3>
<p>The most comprehensive way to explore co-expression patterns is
through ScIsoX’s interactive Shiny application:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Launch interactive co-expression explorer</span></span>
<span><span class="fu"><a href="../reference/launch_coexpression_app.html">launch_coexpression_app</a></span><span class="op">(</span><span class="va">integrated_scht</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="features-of-the-co-expression-app">Features of the Co-expression App<a class="anchor" aria-label="anchor" href="#features-of-the-co-expression-app"></a>
</h3>
<p>The Shiny app provides:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Gene Selection Panel</strong>
<ul>
<li>Search for any gene by name</li>
<li>Filter genes by number of isoforms</li>
<li>Select from pre-defined gene sets</li>
</ul>
</li>
<li>
<strong>Correlation Analysis</strong>
<ul>
<li>Choose between Pearson, Spearman, or Kendall correlation</li>
<li>Adjust minimum cell and expression thresholds</li>
<li>Interactive heatmap with customisable colours</li>
</ul>
</li>
<li>
<strong>Switching Detection</strong>
<ul>
<li>Set correlation thresholds for switching</li>
<li>Visualise switching pairs</li>
<li>Export switching results</li>
</ul>
</li>
<li>
<strong>Cell Type-Specific Analysis</strong>
<ul>
<li>Compare patterns across cell types</li>
<li>Identify cell type-specific switching</li>
<li>Interactive plots for each cell type</li>
</ul>
</li>
<li>
<strong>Export Options</strong>
<ul>
<li>Download correlation matrices</li>
<li>Save publication-ready figures</li>
<li>Export switching pair tables</li>
</ul>
</li>
</ol>
</div>
</div>
<div class="section level2">
<h2 id="programmatic-co-expression-analysis">Programmatic Co-expression Analysis<a class="anchor" aria-label="anchor" href="#programmatic-co-expression-analysis"></a>
</h2>
<div class="section level3">
<h3 id="basic-analysis-example">Basic Analysis Example<a class="anchor" aria-label="anchor" href="#basic-analysis-example"></a>
</h3>
<p>While the Shiny app provides comprehensive interactive analysis, you
can also perform co-expression analysis programmatically:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example: Analyse Mapk13 gene</span></span>
<span><span class="va">target_gene</span> <span class="op">&lt;-</span> <span class="st">"Mapk13"</span></span>
<span></span>
<span><span class="co"># Calculate isoform correlations</span></span>
<span><span class="va">cor_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_isoform_coexpression.html">calculate_isoform_coexpression</a></span><span class="op">(</span></span>
<span>  <span class="va">integrated_scht</span>,</span>
<span>  gene <span class="op">=</span> <span class="va">target_gene</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"pearson"</span>,</span>
<span>  min_cells <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View correlation summary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Gene:"</span>, <span class="va">cor_result</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Gene: Mapk13"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Number of isoforms:"</span>, <span class="va">cor_result</span><span class="op">$</span><span class="va">n_isoforms</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Number of isoforms: 7"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Number of cells:"</span>, <span class="va">cor_result</span><span class="op">$</span><span class="va">n_cells</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Number of cells: 11"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="st">"\nCorrelation matrix:"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "\nCorrelation matrix:"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">cor_result</span><span class="op">$</span><span class="va">cor_matrix</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    ENSMUST00000233984 ENSMUST00000124099 ENSMUST00000129096</span></span>
<span><span class="co">#&gt; ENSMUST00000233984               1.00               0.41               0.19</span></span>
<span><span class="co">#&gt; ENSMUST00000124099               0.41               1.00               0.84</span></span>
<span><span class="co">#&gt; ENSMUST00000129096               0.19               0.84               1.00</span></span>
<span><span class="co">#&gt; ENSMUST00000233109               0.31              -0.10              -0.44</span></span>
<span><span class="co">#&gt; ENSMUST00000233051               0.47              -0.32              -0.66</span></span>
<span><span class="co">#&gt; ENSMUST00000233676               0.78               0.68               0.47</span></span>
<span><span class="co">#&gt; ENSMUST00000004986               0.57               0.39               0.12</span></span>
<span><span class="co">#&gt;                    ENSMUST00000233109 ENSMUST00000233051 ENSMUST00000233676</span></span>
<span><span class="co">#&gt; ENSMUST00000233984               0.31               0.47               0.78</span></span>
<span><span class="co">#&gt; ENSMUST00000124099              -0.10              -0.32               0.68</span></span>
<span><span class="co">#&gt; ENSMUST00000129096              -0.44              -0.66               0.47</span></span>
<span><span class="co">#&gt; ENSMUST00000233109               1.00               0.55               0.13</span></span>
<span><span class="co">#&gt; ENSMUST00000233051               0.55               1.00               0.33</span></span>
<span><span class="co">#&gt; ENSMUST00000233676               0.13               0.33               1.00</span></span>
<span><span class="co">#&gt; ENSMUST00000004986               0.19               0.44               0.73</span></span>
<span><span class="co">#&gt;                    ENSMUST00000004986</span></span>
<span><span class="co">#&gt; ENSMUST00000233984               0.57</span></span>
<span><span class="co">#&gt; ENSMUST00000124099               0.39</span></span>
<span><span class="co">#&gt; ENSMUST00000129096               0.12</span></span>
<span><span class="co">#&gt; ENSMUST00000233109               0.19</span></span>
<span><span class="co">#&gt; ENSMUST00000233051               0.44</span></span>
<span><span class="co">#&gt; ENSMUST00000233676               0.73</span></span>
<span><span class="co">#&gt; ENSMUST00000004986               1.00</span></span>
<span></span>
<span><span class="co"># Detect switching patterns</span></span>
<span><span class="va">switching_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/detect_isoform_switching.html">detect_isoform_switching</a></span><span class="op">(</span></span>
<span>  <span class="va">cor_result</span>,</span>
<span>  threshold <span class="op">=</span> <span class="op">-</span><span class="fl">0.3</span>,</span>
<span>  strong_threshold <span class="op">=</span> <span class="op">-</span><span class="fl">0.5</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"\nNumber of switching pairs:"</span>, <span class="va">switching_result</span><span class="op">$</span><span class="va">n_switching_pairs</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "\nNumber of switching pairs: 3"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Number of strong switching pairs:"</span>, <span class="va">switching_result</span><span class="op">$</span><span class="va">n_strong_switching</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Number of strong switching pairs: 1"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="systematic-analysis">Systematic Analysis<a class="anchor" aria-label="anchor" href="#systematic-analysis"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Analyse multiple genes</span></span>
<span><span class="va">selected_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Atl1"</span>, <span class="st">"Irf8"</span>, <span class="st">"Mx1"</span>, <span class="st">"Mapk13"</span>, <span class="st">"Crb3"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summary of switching patterns</span></span>
<span><span class="va">switching_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">gene</span> <span class="kw">in</span> <span class="va">selected_genes</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/conditions.html" class="external-link">tryCatch</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="va">cor_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_isoform_coexpression.html">calculate_isoform_coexpression</a></span><span class="op">(</span><span class="va">scht_obj</span>, <span class="va">gene</span>, min_cells <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span>    <span class="va">switch_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/detect_isoform_switching.html">detect_isoform_switching</a></span><span class="op">(</span><span class="va">cor_res</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="va">switching_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">switching_summary</span>, <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>      gene <span class="op">=</span> <span class="va">gene</span>,</span>
<span>      n_isoforms <span class="op">=</span> <span class="va">cor_res</span><span class="op">$</span><span class="va">n_isoforms</span>,</span>
<span>      n_switching_pairs <span class="op">=</span> <span class="va">switch_res</span><span class="op">$</span><span class="va">n_switching_pairs</span>,</span>
<span>      n_strong_switching <span class="op">=</span> <span class="va">switch_res</span><span class="op">$</span><span class="va">n_strong_switching</span>,</span>
<span>      stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>    <span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span>, error <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">e</span><span class="op">)</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">switching_summary</span><span class="op">)</span></span>
<span><span class="co">#&gt; data frame with 0 columns and 0 rows</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="cell-type-specific-analysis">Cell Type-Specific Analysis<a class="anchor" aria-label="anchor" href="#cell-type-specific-analysis"></a>
</h2>
<div class="section level3">
<h3 id="understanding-cell-type-patterns">Understanding Cell Type Patterns<a class="anchor" aria-label="anchor" href="#understanding-cell-type-patterns"></a>
</h3>
<p>For cell type-specific co-expression analysis, the Shiny app
provides:</p>
<ul>
<li>
<strong>Side-by-side comparisons</strong> of correlation matrices
across cell types</li>
<li>
<strong>Interactive selection</strong> of cell types to compare</li>
<li>
<strong>Highlighting</strong> of cell type-specific switching
pairs</li>
<li>
<strong>Export</strong> of cell type-specific results</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="advanced-features-in-the-shiny-app">Advanced Features in the Shiny App<a class="anchor" aria-label="anchor" href="#advanced-features-in-the-shiny-app"></a>
</h2>
<div class="section level3">
<h3 id="conservation-analysis-tab">Conservation Analysis Tab<a class="anchor" aria-label="anchor" href="#conservation-analysis-tab"></a>
</h3>
<p>The Shiny app includes a dedicated tab for conservation analysis:</p>
<ul>
<li>
<strong>Multi-gene selection</strong>: Analyse conservation patterns
across multiple genes</li>
<li>
<strong>Threshold adjustment</strong>: Interactively set
conservation thresholds</li>
<li>
<strong>Visualisation options</strong>: Heatmaps, dendrograms, and
network views</li>
<li>
<strong>Export conservation scores</strong>: Download results for
further analysis</li>
</ul>
</div>
<div class="section level3">
<h3 id="network-analysis-tab">Network Analysis Tab<a class="anchor" aria-label="anchor" href="#network-analysis-tab"></a>
</h3>
<p>Explore co-expression networks:</p>
<ul>
<li>
<strong>Network construction</strong>: Build networks based on
correlation thresholds</li>
<li>
<strong>Interactive visualisation</strong>: Drag and zoom network
plots</li>
<li>
<strong>Module detection</strong>: Identify co-expressed isoform
modules</li>
<li>
<strong>Cell type overlays</strong>: Colour networks by cell type
specificity</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="tips-for-using-the-shiny-app">Tips for Using the Shiny App<a class="anchor" aria-label="anchor" href="#tips-for-using-the-shiny-app"></a>
</h2>
<div class="section level3">
<h3 id="getting-started">Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h3>
<ol style="list-style-type: decimal">
<li>
<strong>Launch the app</strong> with your IntegratedSCHT object</li>
<li>
<strong>Start with Overview tab</strong> to see summary
statistics</li>
<li>
<strong>Select a gene</strong> from the dropdown or search box</li>
<li>
<strong>Explore correlations</strong> in the Analysis tab</li>
<li>
<strong>Check cell types</strong> in the Cell Type Specific tab</li>
<li>
<strong>Export results</strong> using the download buttons</li>
</ol>
</div>
<div class="section level3">
<h3 id="performance-tips">Performance Tips<a class="anchor" aria-label="anchor" href="#performance-tips"></a>
</h3>
<ul>
<li>For large datasets, pre-filter to genes of interest</li>
<li>Use the “cache results” option for frequently analysed genes</li>
<li>Export intermediate results to avoid re-computation</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="common-use-cases">Common Use Cases<a class="anchor" aria-label="anchor" href="#common-use-cases"></a>
</h2>
<div class="section level3">
<h3 id="use-case-1-identifying-isoform-switches-during-differentiation">Use Case 1: Identifying Isoform Switches During Differentiation<a class="anchor" aria-label="anchor" href="#use-case-1-identifying-isoform-switches-during-differentiation"></a>
</h3>
<p>The Shiny app is particularly useful for: - Tracking isoform usage
changes across developmental stages - Identifying key switching events -
Comparing patterns between lineages</p>
</div>
<div class="section level3">
<h3 id="use-case-2-disease-associated-isoform-changes">Use Case 2: Disease-Associated Isoform Changes<a class="anchor" aria-label="anchor" href="#use-case-2-disease-associated-isoform-changes"></a>
</h3>
<ul>
<li>Compare normal vs disease samples</li>
<li>Identify disease-specific switching patterns</li>
<li>Explore therapeutic targets</li>
</ul>
</div>
<div class="section level3">
<h3 id="use-case-3-cell-type-marker-discovery">Use Case 3: Cell Type Marker Discovery<a class="anchor" aria-label="anchor" href="#use-case-3-cell-type-marker-discovery"></a>
</h3>
<ul>
<li>Find isoforms specific to cell types</li>
<li>Validate cell type annotations</li>
<li>Discover novel cell type markers</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="troubleshooting-the-shiny-app">Troubleshooting the Shiny App<a class="anchor" aria-label="anchor" href="#troubleshooting-the-shiny-app"></a>
</h2>
<div class="section level3">
<h3 id="common-issues">Common Issues<a class="anchor" aria-label="anchor" href="#common-issues"></a>
</h3>
<ol style="list-style-type: decimal">
<li>
<strong>App won’t launch</strong>
<ul>
<li>Ensure all required packages are installed</li>
<li>Check that your object is an IntegratedSCHT</li>
<li>Verify sufficient memory for your dataset</li>
</ul>
</li>
<li>
<strong>Slow performance</strong>
<ul>
<li>Reduce number of genes in analysis</li>
<li>Increase min_cells threshold</li>
<li>Use sampling for large datasets</li>
</ul>
</li>
<li>
<strong>No results displayed</strong>
<ul>
<li>Check gene names match your data</li>
<li>Verify minimum expression thresholds</li>
<li>Ensure sufficient cells per cell type</li>
</ul>
</li>
</ol>
</div>
</div>
<div class="section level2">
<h2 id="exporting-results-from-the-shiny-app">Exporting Results from the Shiny App<a class="anchor" aria-label="anchor" href="#exporting-results-from-the-shiny-app"></a>
</h2>
<p>The app provides multiple export options:</p>
<div class="section level3">
<h3 id="available-export-formats">Available Export Formats<a class="anchor" aria-label="anchor" href="#available-export-formats"></a>
</h3>
<ol style="list-style-type: decimal">
<li>
<strong>Correlation Matrices</strong>
<ul>
<li>CSV format for further analysis</li>
<li>RDS format to preserve R object structure</li>
</ul>
</li>
<li>
<strong>Switching Results</strong>
<ul>
<li>Table of switching pairs</li>
<li>Summary statistics</li>
</ul>
</li>
<li>
<strong>Visualisations</strong>
<ul>
<li>High-resolution PNG/PDF</li>
<li>Customisable dimensions</li>
<li>Publication-ready figures</li>
</ul>
</li>
<li>
<strong>Complete Analysis Reports</strong>
<ul>
<li>HTML reports with all results</li>
<li>Includes methods and parameters</li>
<li>Reproducible analysis record</li>
</ul>
</li>
</ol>
</div>
<div class="section level3">
<h3 id="programmatic-export">Programmatic Export<a class="anchor" aria-label="anchor" href="#programmatic-export"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># After running analysis in the app, you can also export programmatically</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/exists.html" class="external-link">exists</a></span><span class="op">(</span><span class="st">"ct_coexp"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Export to temporary directory</span></span>
<span>  <span class="va">export_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/export_coexpression_results.html">export_coexpression_results</a></span><span class="op">(</span></span>
<span>    <span class="va">ct_coexp</span>,</span>
<span>    output_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    gene_name <span class="op">=</span> <span class="st">"Atl1"</span>,</span>
<span>    format <span class="op">=</span> <span class="st">"csv"</span></span>
<span>  <span class="op">)</span></span>
<span>  </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Results exported to:"</span>, <span class="va">export_result</span><span class="op">$</span><span class="va">output_dir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>The ScIsoX Shiny application provides a comprehensive, user-friendly
interface for co-expression analysis. Whether you’re exploring a single
gene or conducting systematic analyses across cell types, the app
streamlines the workflow and provides publication-ready outputs.</p>
<p>For more advanced programmatic analysis, the underlying functions
(<code>calculate_isoform_coexpression</code>,
<code>detect_isoform_switching</code>, etc.) remain available for custom
workflows and batch processing.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.3 (2025-02-28)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS 26.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Australia/Brisbane</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ggplot2_4.0.0 ScIsoX_1.1.3 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] bitops_1.0-9                gridExtra_2.3              </span></span>
<span><span class="co">#&gt;   [3] rlang_1.1.6                 magrittr_2.0.4             </span></span>
<span><span class="co">#&gt;   [5] clue_0.3-66                 GetoptLong_1.0.5           </span></span>
<span><span class="co">#&gt;   [7] ggridges_0.5.6              matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt;   [9] compiler_4.4.3              png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [11] systemfonts_1.2.1           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;  [13] pkgconfig_2.0.3             shape_1.4.6.1              </span></span>
<span><span class="co">#&gt;  [15] crayon_1.5.3                fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [17] XVector_0.46.0              promises_1.3.3             </span></span>
<span><span class="co">#&gt;  [19] Rsamtools_2.22.0            rmarkdown_2.29             </span></span>
<span><span class="co">#&gt;  [21] UCSC.utils_1.2.0            ragg_1.4.0                 </span></span>
<span><span class="co">#&gt;  [23] purrr_1.1.0                 xfun_0.53                  </span></span>
<span><span class="co">#&gt;  [25] zlibbioc_1.52.0             cachem_1.1.0               </span></span>
<span><span class="co">#&gt;  [27] GenomeInfoDb_1.42.3         jsonlite_2.0.0             </span></span>
<span><span class="co">#&gt;  [29] progress_1.2.3              later_1.4.4                </span></span>
<span><span class="co">#&gt;  [31] DelayedArray_0.32.0         BiocParallel_1.40.0        </span></span>
<span><span class="co">#&gt;  [33] parallel_4.4.3              prettyunits_1.2.0          </span></span>
<span><span class="co">#&gt;  [35] cluster_2.1.8.1             R6_2.6.1                   </span></span>
<span><span class="co">#&gt;  [37] bslib_0.9.0                 RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt;  [39] rtracklayer_1.66.0          car_3.1-3                  </span></span>
<span><span class="co">#&gt;  [41] GenomicRanges_1.58.0        jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [43] diptest_0.77-2              Rcpp_1.1.0                 </span></span>
<span><span class="co">#&gt;  [45] SummarizedExperiment_1.36.0 iterators_1.0.14           </span></span>
<span><span class="co">#&gt;  [47] knitr_1.50                  IRanges_2.40.1             </span></span>
<span><span class="co">#&gt;  [49] httpuv_1.6.16               Matrix_1.7-3               </span></span>
<span><span class="co">#&gt;  [51] tidyselect_1.2.1            viridis_0.6.5              </span></span>
<span><span class="co">#&gt;  [53] rstudioapi_0.17.1           abind_1.4-8                </span></span>
<span><span class="co">#&gt;  [55] yaml_2.3.10                 miniUI_0.1.1.1             </span></span>
<span><span class="co">#&gt;  [57] doParallel_1.0.17           codetools_0.2-20           </span></span>
<span><span class="co">#&gt;  [59] curl_7.0.0                  lattice_0.22-6             </span></span>
<span><span class="co">#&gt;  [61] tibble_3.3.0                plyr_1.8.9                 </span></span>
<span><span class="co">#&gt;  [63] withr_3.0.2                 shiny_1.11.1               </span></span>
<span><span class="co">#&gt;  [65] Biobase_2.66.0              S7_0.2.0                   </span></span>
<span><span class="co">#&gt;  [67] evaluate_1.0.5              moments_0.14.1             </span></span>
<span><span class="co">#&gt;  [69] desc_1.4.3                  circlize_0.4.16            </span></span>
<span><span class="co">#&gt;  [71] mclust_6.1.1                Biostrings_2.74.1          </span></span>
<span><span class="co">#&gt;  [73] pillar_1.11.1               MatrixGenerics_1.18.1      </span></span>
<span><span class="co">#&gt;  [75] carData_3.0-5               DT_0.34.0                  </span></span>
<span><span class="co">#&gt;  [77] foreach_1.5.2               stats4_4.4.3               </span></span>
<span><span class="co">#&gt;  [79] plotly_4.11.0               generics_0.1.4             </span></span>
<span><span class="co">#&gt;  [81] RCurl_1.98-1.16             S4Vectors_0.44.0           </span></span>
<span><span class="co">#&gt;  [83] hms_1.1.3                   scales_1.4.0               </span></span>
<span><span class="co">#&gt;  [85] xtable_1.8-4                glue_1.8.0                 </span></span>
<span><span class="co">#&gt;  [87] lazyeval_0.2.2              tools_4.4.3                </span></span>
<span><span class="co">#&gt;  [89] BiocIO_1.16.0               data.table_1.17.8          </span></span>
<span><span class="co">#&gt;  [91] ggradar_0.2                 GenomicAlignments_1.42.0   </span></span>
<span><span class="co">#&gt;  [93] fs_1.6.6                    XML_3.99-0.18              </span></span>
<span><span class="co">#&gt;  [95] cowplot_1.2.0               grid_4.4.3                 </span></span>
<span><span class="co">#&gt;  [97] tidyr_1.3.1                 colorspace_2.1-2           </span></span>
<span><span class="co">#&gt;  [99] patchwork_1.3.0             GenomeInfoDbData_1.2.13    </span></span>
<span><span class="co">#&gt; [101] restfulr_0.0.15             Formula_1.2-5              </span></span>
<span><span class="co">#&gt; [103] cli_3.6.5                   textshaping_1.0.0          </span></span>
<span><span class="co">#&gt; [105] S4Arrays_1.6.0              viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt; [107] ComplexHeatmap_2.22.0       dplyr_1.1.4                </span></span>
<span><span class="co">#&gt; [109] gtable_0.3.6                sass_0.4.10                </span></span>
<span><span class="co">#&gt; [111] digest_0.6.37               BiocGenerics_0.52.0        </span></span>
<span><span class="co">#&gt; [113] ggrepel_0.9.6               SparseArray_1.6.2          </span></span>
<span><span class="co">#&gt; [115] rjson_0.2.23                htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt; [117] farver_2.1.2                htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt; [119] pkgdown_2.1.3               lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt; [121] httr_1.4.7                  mime_0.13                  </span></span>
<span><span class="co">#&gt; [123] GlobalOptions_0.1.2         ggExtra_0.10.1             </span></span>
<span><span class="co">#&gt; [125] MASS_7.3-65</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Siyuan Wu, Ulf Schmitz<br> Site built with <a href="https://pkgdown.r-lib.org" class="external-link">pkgdown</a> 2.1.1</p>
</div>

<div class="pkgdown-footer-right">
  <p><strong>Package License</strong>: <a href="https://opensource.org/licenses/MIT" class="external-link">MIT</a><br><br><strong>Example Data Attribution</strong>:<br> The example datasets included in this package are derived from:<br> Wang et al. (2022). Single-cell architecture and functional requirement of alternative splicing during hematopoietic stem cell formation. <em>Science Advances</em> 8(1):eabg5369<br> Licensed under <a href="https://creativecommons.org/licenses/by/4.0/" class="external-link">CC BY 4.0</a><br> We gratefully acknowledge the authors for making their data publicly available.</p>
</div>

    </footer>
</div>





  </body>
</html>
