<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Calculate isoform co-expression correlation matrix — calculate_isoform_coexpression • ScIsoX</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Calculate isoform co-expression correlation matrix — calculate_isoform_coexpression"><meta name="description" content="Calculates correlation matrix between isoforms of a gene using
the specified correlation method."><meta property="og:description" content="Calculates correlation matrix between isoforms of a gene using
the specified correlation method."><meta property="og:image" content="https://ThaddeusWu.github.io/ScIsoX/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-#1a5490" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ScIsoX</a>

    <small class="nav-text text-success me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.1.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><a class="dropdown-item" href="../articles/vignette-01-getting-started.html">Getting Started</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-02-data-import-qc.html">Data Import and QC</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-03-complexity-metrics.html">Complexity Metrics</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-04-visualisation.html">Visualisation Gallery</a></li>
    <li><a class="dropdown-item" href="../articles/vignette-05-coexpression.html">Co-expression Analysis</a></li>
  </ul></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-news" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">News</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-news"><li><h6 class="dropdown-header" data-toc-skip>Releases</h6></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/v1.1.3">Version 1.1.3</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/v1.1.2">Version 1.1.2</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/v1.1.1">Version 1.1.1</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/ThaddeusWu/ScIsoX/releases/tag/V1.1.0">Version 1.1.0</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../news/index.html">Changelog</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ThaddeusWu/ScIsoX/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Calculate isoform co-expression correlation matrix</h1>
      <small class="dont-index">Source: <a href="https://github.com/ThaddeusWu/ScIsoX/blob/HEAD/R/coexpression_analysis.R" class="external-link"><code>R/coexpression_analysis.R</code></a></small>
      <div class="d-none name"><code>calculate_isoform_coexpression.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Calculates correlation matrix between isoforms of a gene using
the specified correlation method.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">calculate_isoform_coexpression</span><span class="op">(</span></span>
<span>  <span class="va">scht_obj</span>,</span>
<span>  <span class="va">gene</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"pearson"</span>,</span>
<span>  min_cells <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  min_expression <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-scht-obj">scht_obj<a class="anchor" aria-label="anchor" href="#arg-scht-obj"></a></dt>
<dd><p>SCHT or IntegratedSCHT object</p></dd>


<dt id="arg-gene">gene<a class="anchor" aria-label="anchor" href="#arg-gene"></a></dt>
<dd><p>Gene name to analyse</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p>Correlation method: "pearson", "spearman", or "kendall"</p></dd>


<dt id="arg-min-cells">min_cells<a class="anchor" aria-label="anchor" href="#arg-min-cells"></a></dt>
<dd><p>Minimum number of cells required for analysis</p></dd>


<dt id="arg-min-expression">min_expression<a class="anchor" aria-label="anchor" href="#arg-min-expression"></a></dt>
<dd><p>Minimum mean expression threshold</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list containing:</p><ul><li><p>cor_matrix: Correlation matrix between isoforms</p></li>
<li><p>n_isoforms: Number of isoforms</p></li>
<li><p>n_cells: Number of cells</p></li>
<li><p>method: Correlation method used</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Load example data and create SCHT</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_counts_blood</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">transcript_counts_blood</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">transcript_info</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">sample2stage</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">scht_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="create_scht.html">create_scht</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  gene_counts <span class="op">=</span> <span class="va">gene_counts_blood</span>,</span></span>
<span class="r-in"><span>  transcript_counts <span class="op">=</span> <span class="va">transcript_counts_blood</span>,</span></span>
<span class="r-in"><span>  transcript_info <span class="op">=</span> <span class="va">transcript_info</span>,</span></span>
<span class="r-in"><span>  cell_info <span class="op">=</span> <span class="va">sample2stage</span>,</span></span>
<span class="r-in"><span>  qc_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    min_genes_per_cell <span class="op">=</span> <span class="fl">4000</span>,       </span></span>
<span class="r-in"><span>    max_genes_per_cell <span class="op">=</span> <span class="fl">10000</span>,      </span></span>
<span class="r-in"><span>    min_cells_expressing <span class="op">=</span> <span class="fl">0.02</span>,   </span></span>
<span class="r-in"><span>    min_expr <span class="op">=</span> <span class="fl">1e-6</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  n_hvg <span class="op">=</span> <span class="fl">3000</span>,</span></span>
<span class="r-in"><span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="calculate_isoform_coexpression-1.png" alt="" width="768" height="576"></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> === Sparsity Analysis for Current Dataset ===</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1. Original Transcript Matrix:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Number of genes: 55,487</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Number of isoforms: 142,238</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Number of cells: 205</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Non-zero elements: 2,457,789</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Zero elements: 26,701,001</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Total elements: 29,158,790</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Sparsity: 91.57%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2. Filtered Transcript Matrix (Post-QC HVG):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Number of genes: 1,936</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Number of isoforms: 7,327</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Number of cells: 181</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Non-zero elements: 73,057</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Zero elements: 1,253,130</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Total elements: 1,326,187</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Sparsity: 94.49%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3. SCHT Structure (Post-QC HVG):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Non-zero elements: 73,057</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Zero elements: 154,780</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Total elements: 227,837</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Sparsity: 67.93%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4. Naive 3D Tensor (Post-QC HVG):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Required dimensions: 1,936 x 21 x 181</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Non-zero elements: 73,057</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Zero elements: 7,285,679</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Total elements: 7,358,736</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Sparsity: 99.01%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5. Zero Elements Avoided by SCHT:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   vs Original Matrix: 26,546,221</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   vs Filtered Matrix: 1,098,350</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   vs Naive 3D Tensor: 7,130,899</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6. Memory Efficiency Summary:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   SCHT uses only 0.78% of original matrix size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   SCHT uses only 17.18% of filtered matrix size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   SCHT uses only 3.10% of naive tensor size</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Example 1: Find genes with multiple isoforms in the filtered data</span></span></span>
<span class="r-in"><span><span class="co"># Get the actual gene list from the SCHT object</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">inherits</a></span><span class="op">(</span><span class="va">scht_obj</span>, <span class="st">"IntegratedSCHT"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>  <span class="va">gene_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">scht_obj</span><span class="op">$</span><span class="va">original_results</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span></span>
<span class="r-in"><span>  <span class="va">gene_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">scht_obj</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Find genes with multiple isoforms</span></span></span>
<span class="r-in"><span><span class="va">multi_iso_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw">for</span> <span class="op">(</span><span class="va">g</span> <span class="kw">in</span> <span class="va">gene_list</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gene_list</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">inherits</a></span><span class="op">(</span><span class="va">scht_obj</span>, <span class="st">"IntegratedSCHT"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">n_iso</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">scht_obj</span><span class="op">$</span><span class="va">original_results</span><span class="op">[[</span><span class="va">g</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">n_iso</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">scht_obj</span><span class="op">[[</span><span class="va">g</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">}</span></span></span>
<span class="r-in"><span>  <span class="kw">if</span> <span class="op">(</span><span class="va">n_iso</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">)</span> <span class="va">multi_iso_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">multi_iso_genes</span>, <span class="va">g</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Analyze the first multi-isoform gene found</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">multi_iso_genes</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>  <span class="va">cor_result</span> <span class="op">&lt;-</span> <span class="fu">calculate_isoform_coexpression</span><span class="op">(</span><span class="va">scht_obj</span>, <span class="va">multi_iso_genes</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Gene:"</span>, <span class="va">multi_iso_genes</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="st">"\n"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Number of isoforms:"</span>, <span class="va">cor_result</span><span class="op">$</span><span class="va">n_isoforms</span>, <span class="st">"\n"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">cor_result</span><span class="op">$</span><span class="va">cor_matrix</span><span class="op">)</span></span></span>
<span class="r-in"><span>  </span></span>
<span class="r-in"><span>  <span class="co"># Check switching patterns</span></span></span>
<span class="r-in"><span>  <span class="va">switching</span> <span class="op">&lt;-</span> <span class="fu"><a href="detect_isoform_switching.html">detect_isoform_switching</a></span><span class="op">(</span><span class="va">cor_result</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="kw">if</span> <span class="op">(</span><span class="va">switching</span><span class="op">$</span><span class="va">n_switching_pairs</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Switching pairs found:"</span>, <span class="va">switching</span><span class="op">$</span><span class="va">n_switching_pairs</span>, <span class="st">"\n"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">}</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Gene: Chil3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Number of isoforms: 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    ENSMUST00000063062 ENSMUST00000149836 ENSMUST00000129563</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUST00000063062          1.0000000          0.8750332          0.9992373</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUST00000149836          0.8750332          1.0000000          0.8733112</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSMUST00000129563          0.9992373          0.8733112          1.0000000</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Siyuan Wu, Ulf Schmitz<br> Site built with <a href="https://pkgdown.r-lib.org" class="external-link">pkgdown</a> 2.1.1</p>
</div>

<div class="pkgdown-footer-right">
  <p><strong>Package License</strong>: <a href="https://opensource.org/licenses/MIT" class="external-link">MIT</a><br><br><strong>Example Data Attribution</strong>:<br> The example datasets included in this package are derived from:<br> Wang et al. (2022). Single-cell architecture and functional requirement of alternative splicing during hematopoietic stem cell formation. <em>Science Advances</em> 8(1):eabg5369<br> Licensed under <a href="https://creativecommons.org/licenses/by/4.0/" class="external-link">CC BY 4.0</a><br> We gratefully acknowledge the authors for making their data publicly available.</p>
</div>

    </footer></div>





  </body></html>

